{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "f433c883-ba97-4bbe-bedd-5153db6b8cb6",
   "metadata": {},
   "outputs": [],
   "source": [
    "from NER_processing import nlp, disease_service, chemical_service, genetic_service\n",
    "import pysolr"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6d2036d9-77c1-4234-9b6f-42dcce8125fb",
   "metadata": {},
   "source": [
    "# Replicate paragraphs in local"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "b09257a3-57a4-49d4-b6a6-b13b72a6a40a",
   "metadata": {},
   "outputs": [],
   "source": [
    "solr = pysolr.Solr('http://librairy.linkeddata.es/data/covid-paragraphs', timeout=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d204dd18-d5bf-4f6f-9d02-47a26b9ea3fa",
   "metadata": {},
   "outputs": [],
   "source": [
    "solr_local = pysolr.Solr('http://localhost:8984/solr/covid_paragraphs', timeout=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "a0d07f48-9bc4-4922-a53c-04553fd17853",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'{\\n  \"responseHeader\":{\\n    \"zkConnected\":null,\\n    \"status\":0,\\n    \"QTime\":1,\\n    \"params\":{\\n      \"q\":\"{!lucene}*:*\",\\n      \"distrib\":\"false\",\\n      \"df\":\"_text_\",\\n      \"rows\":\"10\",\\n      \"echoParams\":\"all\",\\n      \"rid\":\"-3\"}},\\n  \"status\":\"OK\"}\\n'"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "solr_local.ping()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "963d1146-3fe4-421a-8d79-cbb3ee3d1f1c",
   "metadata": {},
   "outputs": [],
   "source": [
    "results = solr.search(q=\"*:*\", rows=60000,sort=\"id asc\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "47856712-d224-46f3-9c4a-d69af278c28e",
   "metadata": {},
   "outputs": [],
   "source": [
    "j=0\n",
    "paragraphs = []\n",
    "for result in results:\n",
    "    paragraph = {}\n",
    "    if ('section_s' in result):\n",
    "        paragraph['section_s'] = result['section_s']\n",
    "    if ('id' in result):\n",
    "        paragraph['id'] = result['id']\n",
    "    if ('article_id_s' in result):\n",
    "        paragraph['article_id_s'] = result['article_id_s']\n",
    "    if ('text_t' in result):\n",
    "        paragraph['text_t'] = result['text_t']\n",
    "    if ('size_i' in result):\n",
    "        paragraph['size_i'] = result['size_i']\n",
    "    if ('name_s' in result):\n",
    "        paragraph['name_s'] = result['name_s']\n",
    "    paragraphs.append(paragraph)\n",
    "    j+=1\n",
    "    if j%10000 == 0:\n",
    "        solr_local.add(paragraphs)\n",
    "        print(j, 'paragraphs added')\n",
    "        paragraphs = []"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "86056f14-65e9-4830-9d27-b700d065c478",
   "metadata": {},
   "source": [
    "# Annotate in local"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "819acc28-fd89-4ecc-b68c-4fb1ed292220",
   "metadata": {},
   "outputs": [],
   "source": [
    "import time"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "2f543dde-f365-433c-b80d-f5b9d8ff2db3",
   "metadata": {},
   "outputs": [],
   "source": [
    "solr_local = pysolr.Solr('http://localhost:8984/solr/covid_paragraphs', always_commit=True, timeout=50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "f3872f77-17a4-4f99-85fa-9fae2449d344",
   "metadata": {},
   "outputs": [],
   "source": [
    "paragraphs = solr_local.search(q='*:*', rows=200, start=200, sort=\"id ASC\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "2e006f41-36a0-4088-9510-74dfe1d3b652",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "200\n"
     ]
    }
   ],
   "source": [
    "print(len(paragraphs))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "88c295bf-d222-448b-9408-c10d97bfd52d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "reading from solr..\n",
      "100 docs evaluated\n",
      "{'id': '0000017c2479d436783516b1fe7ff927', 'section_s': 'disease spreading in a quickly adaptive network structure', 'article_id_s': '6e48dbcd16f8a5ac136397a92e6a6bad4d52a6bb', 'size_i': 122, 'name_s': 'disease spreading in a quickly adaptive network structure', 'text_t': ['This quantity exceeds zero whenever Note that taking r D 0 yields the basic reproductive ratio R A 0 for both SIR and SIS:'], 'disease_ents': [], 'chemical_ents': [], 'genetic_ents': []}\n",
      "100 paragraphs annotated\n",
      "200 docs evaluated\n",
      "{'id': '0002d778d037e737863800a22cc88f06', 'section_s': 'reference', 'article_id_s': 'e5d54917d149ed87572748aafa6b0847668ce5a6', 'size_i': 117, 'name_s': 'reference', 'text_t': ['Hiestand BC, Smith SW. Cocaine chest pain: between a (crack) rock and a hard place. Acad Emerg Med. 2011;18(1):68-71.'], 'disease_ents': ['chest pain'], 'chemical_ents': ['Cocaine'], 'genetic_ents': []}\n",
      "200 paragraphs annotated\n",
      "300 docs evaluated\n",
      "{'id': '000601bccdd22fa74af9a8a908c1e9e0', 'section_s': 'toll-like receptor adaptor trif has a protective role in the host response to sars-cov. ::: results', 'article_id_s': 'PMC4447251', 'size_i': 2234, 'name_s': 'toll-like receptor adaptor trif has a protective role in the host response to sars-cov. ::: results', 'text_t': ['Due to the differences in weight loss and survival between TRIF−/− and TLR3−/− mice infected with SARS-CoV, we hypothesized that TLR4 may also contribute to signaling through TRIF in response to SARS-CoV infection. To determine the role of TLR4 in the pathogenesis of SARS-CoV, we infected TLR4−/− mice and wild-type C57BL6/J mice intranasally with 105 PFU of rMA15–SARS-CoV. TLR4−/− mice lost a greater percentage of their starting weight, with statistically significant differences in weight loss compared to wild-type mice on days 3 to 7 postinfection, but began recovering by 7 days postinfection (Fig. 5A; *, P < 0.05; ***, P < 0.001). TLR4−/− mice had significantly higher titers of virus in the lungs than wild-type mice infected with SARS-CoV at days 2 (3-fold difference; ***, P < 0.001) and 4 (4-fold difference; *, P < 0.05) postinfection but had cleared the virus by day 7 postinfection, similarly to wild-type mice (Fig. 5C). Because TLR4 can signal in either a MyD88-dependent or TRIF-dependent manner via the usage of the sorting adaptor MAL or TRAM, respectively, we infected 10-week-old female TRAM−/− mice and wild-type C57BL6/J mice intranasally with 105 PFU of SARS-CoV in order to discriminate between the effects of MyD88-dependent and TRIF-dependent signaling. TRAM−/− mice infected with SARS-CoV lost significantly more weight than wild-type mice (Fig. 5B; ***, P < 0.001). TRAM−/− mice infected with SARS-CoV demonstrated prolonged weight loss through 10 days postinfection, while wild-type mice recovered from weight loss by day 6 postinfection (Fig. 5B). TRAM−/− mice had titers that were 3-fold higher than those seen with wild-type mice infected with SARS-CoV on day 2 postinfection (**, P < 0.01) but not on day 4 postinfection and cleared the virus by day 7 postinfection similarly to wild-type mice, despite lack of recovery from weight loss (Fig. 5D). The alterations in lung function measured by whole-body plethysmography that were observed in TLR4−/− and TRAM−/− mice infected with SARS-CoV were similar to those observed in wild-type mice (see Fig. S3 in the supplemental material), supporting our hypothesis that TLR4/TRAM/TRIF signaling has a modest protective effect against SARS-CoV infection.'], 'disease_ents': ['weight loss', 'SARS-CoV', 'SARS-CoV infection', 'SARS-CoV', 'SARS-CoV.', 'weight loss', 'SARS-CoV', 'SARS-CoV', 'SARS-CoV', 'SARS-CoV', 'weight loss', 'weight loss', 'SARS-CoV', 'SARS-CoV', 'SARS-CoV infection'], 'chemical_ents': [], 'genetic_ents': ['TLR4', 'TRIF', 'TLR4', 'TLR4', 'MyD88', 'TRIF', 'TRAM', 'MyD88', 'TRIF']}\n",
      "300 paragraphs annotated\n",
      "400 docs evaluated\n",
      "{'id': '0008d3d1aee5ab71cd40b3a111318d65', 'section_s': 'structural mapping of disease-related gne mutations', 'article_id_s': '178363fb71ad0ae91e621e6e51fc03bb521f5695', 'size_i': 525, 'name_s': 'structural mapping of disease-related gne mutations', 'text_t': ['The third group currently includes residue P511. P511 has the highest relative solvent accessibility (.40%) among the 23 mutation sites and is located on a loop region of the structure. The underlying mechanism for the association of P511H and P511L mutations with HIBM is elusive without further data, but mutation of a proline to any other residue type will inevitably change the flexibility of the loop region around this residue and could thus change the allostery of full length GNE in the higher-order oligomeric state.'], 'disease_ents': ['HIBM'], 'chemical_ents': [], 'genetic_ents': []}\n",
      "400 paragraphs annotated\n",
      "500 docs evaluated\n",
      "{'id': '000c2df5fd82f0110781c2c117e4ad76', 'section_s': 'could the authors provide any information on the completeness of the data? we addresses available for all cases? were all addresses geocoded (lat/long) successfully?', 'article_id_s': '6a7cec5919100b5ee95c8c84b99455a7611bc79c', 'size_i': 660, 'name_s': 'could the authors provide any information on the completeness of the data? we addresses available for all cases? were all addresses geocoded (lat/long) successfully?', 'text_t': ['All other cases (n=85) were geocoded to caserío (hamlet) level using a combination of open street map, google maps, bing maps and websites listing georeferences of caseríos/ lotificaciones /barrios (small collections of dwellings e.g. hamlet, housing development, neighbourhood) in El Salvador, organised by municipality. We cross checked locations and used fuller address information e.g. descriptors or street names when possible to avoid mis-identification. We have added a section to the supplementary information detailing the online resources used to georeferenced the residential addresses of cases (lines 34 to 45 of Supplementary Information, Note 2) '], 'disease_ents': [], 'chemical_ents': [], 'genetic_ents': []}\n",
      "500 paragraphs annotated\n",
      "600 docs evaluated\n",
      "{'id': '000f9932e507cf9380fedf6f0f862cf2', 'section_s': 'ciliostasis test', 'article_id_s': 'dab074e75c2dae02b60e8094aa5a06e553122831', 'size_i': 1359, 'name_s': 'ciliostasis test', 'text_t': [\"Assessment of protection from the challenge viruses were carried out as described by others (Cook and others 1999) . Five days after challenge, birds were humanely killed, a section of trachea (immediately after larynx to thoracic inlet) was carefully dissected out and placed into warm (37°C) tracheal organ culture medium (Eagle's serum-free minimum essential medium with glutamine and streptomycin (50 mg/ml) and penicillin (50 iu/ml)). The trachea was cut using a tissue chopper (The Mickle Laboratory Engineering) to give 6 µm thickness of rings, and rings of three from the top and bottom, and four from the middle trachea were examined for ciliary activity using a low-power microscope (×100 magnification). For each of the rings, the ciliary activity was scored; 0, all cilia beating; 1, 75 per cent beating; 2, 50 per cent beating; 3, 25 per cent beating; and 4, none beating (100 per cent ciliostasis). For each bird, out of the 10 rings examined, the maximum possible ciliary score is 40, which indicates a total lack of protection (no cilia beating in all 10 rings). The mean ciliary scores for each bird was calculated and percentage protection for each group using a formula described by others (Cook and others 1999) ; [1−(mean ciliostasis score for vaccinated/challenge group)/mean ciliostasis score for corresponding challenge controls)]×100.\"], 'disease_ents': ['ciliostasis', 'ciliostasis', 'ciliostasis'], 'chemical_ents': ['glutamine', 'streptomycin', 'penicillin'], 'genetic_ents': []}\n",
      "600 paragraphs annotated\n",
      "700 docs evaluated\n",
      "{'id': '0012c6a9dbc7b782f939712f4c346ce7', 'section_s': 'discussion', 'article_id_s': 'fbdc0117c66b252ed77172829bf245b090ca83f7', 'size_i': 1972, 'name_s': 'discussion', 'text_t': ['In Scandinavian countries including Denmark, Sweden, Norway, and Finland, BVD had been eradicated due to implementation of an eradication program without vaccination [24, 30] . Moreover, the Norwegian government banned the use of the vaccination for pestiviruses as the vaccination could prevent the manifestation of severe clinical manifestations, but did not prevent the spread of the virus [24] . In these countries, cattle with antibodies against BVDV as potential risk of BVD spread is easily monitored but it can also cull no-virus-shedding cattle which was transiently infected in past. In Japan, on the other hand, we are monitoring virusshedding cattle and adopt the vaccination so as to prevent the further virus spread. According to the results of the present study, it was revealed that the adoption of mass vaccination decreased the number of missed BVD cattle, although it did not increase the number of detected BVD cattle. This may indicate that, in an area where no interventions for BVD control have been implemented, the adoption of the mass vaccination campaign would be the foremost intervention for BVD control to reduce the number of cattle infected with BVDV. In the present model, herd immunity conferred by mass vaccination did not affect the number of PI cattle, but did affect the number of TI cattle due to a decrease of the sensitive population. Under system of no BVD monitoring, PI cattle would not be detected without showing obvious clinical manifestations, such as reproductive disorder. Given these considerations, the missed BVD cattle under the mass vaccination program would be the TI cattle of any age or PI heifers. As these diseased cattle usually do not show obvious clinical manifestations, it is likely that undetected cattle would be left for a period, thereby allowing the shedding of infectious viruses. Even TI cattle would be a source of BVDV spread, resulting in horizontal and vertical infection over a limited duration.'], 'disease_ents': ['BVD', 'pestiviruses', 'BVD', 'BVD', 'BVD', 'BVD', 'BVD', 'BVDV', 'PI', 'TI', 'BVD', 'PI', 'reproductive disorder', 'BVD', 'TI', 'TI', 'infection'], 'chemical_ents': [], 'genetic_ents': []}\n",
      "700 paragraphs annotated\n",
      "800 docs evaluated\n",
      "{'id': '001654cd30abc21267de2b240b3391b3', 'section_s': 'a metabolomics approach to the differentiation of cognitive states in hiv-infected patients ::: p51', 'article_id_s': 'PMC3713197', 'size_i': 48, 'name_s': 'a metabolomics approach to the differentiation of cognitive states in hiv-infected patients ::: p51', 'text_t': ['\\n3Department of Chemistry, University of Oxford;'], 'disease_ents': [], 'chemical_ents': [], 'genetic_ents': []}\n",
      "800 paragraphs annotated\n",
      "900 docs evaluated\n",
      "{'id': '00196c005ea5d694a3c41dd9909682fc', 'section_s': 'body', 'article_id_s': '9f14604a20b4a86b19cf41701f4762005b0ab9eb', 'size_i': 1654, 'name_s': 'body', 'text_t': ['To determine RPA specificity, 10 ng of viral DNA or cDNA was amplified by RPA. As shown in Figure 3A , genomic DNA from CPV-2a and -2b was amplified, while templates from some important dog viruses, i.e., CDV, PRV and CCoV, were not (Fig. 3B ). Viral DNA extracted from 57 clinical fecal samples was tested by RPA, and the results were compared with those obtained using conventional PCR. We observed that 50 out of 57 (83.3 %) Of the seven samples that were negative by PCR, three were negative and four were positive when tested by RPA. The overall diagnostic agreement between RPA and PCR was 92.9 % (53/57). Kumar et al. reported that real-time PCR had a detection limit of 10 copies of CPV-2 genomic DNA/reaction [8] , comparable to our method. We designed the RPA assay aiming to detect all CPV-2 variants. Indeed, both CPV-2a and -2b were successfully amplified with RPA (Fig. 3A) . We were unable to obtain CPV-2c, which was thus not tested in this study. Nevertheless, the RPA primers were 100 % identical to the target sequences in CPV-2c genomic DNA, as shown by sequence alignment. RPA did not amplify the template of other important dog viruses, demonstrating high specificity. The validation experiment revealed that RPA had an excellent diagnostic agreement rate with PCR, which is indicative of the clinical applicability of RPA. It is known that low-temperature amplification usually creates a high background signal; hence, nonspecific bands generated from RPA are not uncommon [17] . However, we also observed that nonspecifically amplified bands can be easily differentiated by size difference and signal intensity (Figures 1, 2, 3) .'], 'disease_ents': ['primers'], 'chemical_ents': [], 'genetic_ents': ['-2b']}\n",
      "900 paragraphs annotated\n",
      "1000 docs evaluated\n",
      "{'id': '001c54c681d92a359a0abd01ec69231a', 'section_s': 'virus challenge of juvenile trout', 'article_id_s': '1b3f9daeb1b6fa660774ccef8f9e545ceaf7bdc9', 'size_i': 449, 'name_s': 'virus challenge of juvenile trout', 'text_t': ['Unknown replicating agents are occasionally detected during routine operations at fish farms or hatcheries. Shortly after virus isolation, these two unknown replicating agents, RbtOV-1 and RbtOV-2, were injected (50 L) into groups of juvenile rainbow trout (about 1 g) with fluid from actively replicating CHSE-214 cell cultures or injected with only uninfected cell culture fluid. Fish were monitored for mortality and observed for gross pathology.'], 'disease_ents': [], 'chemical_ents': [], 'genetic_ents': []}\n",
      "1000 paragraphs annotated\n",
      "1100 docs evaluated\n",
      "{'id': '001f105427b2ac78b67baa936eb0d926', 'section_s': 'institutional background', 'article_id_s': '254dba768fceec0f2a61c2c8a8486f1298ab1b93', 'size_i': 710, 'name_s': 'institutional background', 'text_t': [\"The Convention did not stipulate any particular form of international service structure but rather established national sovereignty over airspace and an institutional framework within which nations could essentially trade these rights (''freedoms of the skies'') amongst themselves. The outcome was a mass of bilateral agreements between countries that, in general, stipulated which airlines that could fly between them, the capacity of each airline, the fares to be charged, and, often, how the revenues generated were to be shared between the carriers. Similarly, air navigation services were sovereign responsibilities and a patchwork of systems emerged that impinged upon any notion of seamless air travel.\"], 'disease_ents': [], 'chemical_ents': [], 'genetic_ents': []}\n",
      "1100 paragraphs annotated\n",
      "1200 docs evaluated\n",
      "{'id': '0022802febce6ce5f0c647fd6c722020', 'section_s': 'conclusions', 'article_id_s': 'b83b223d7e9170483710f5275f46cd578730990f', 'size_i': 1386, 'name_s': 'conclusions', 'text_t': ['The lived experiences and factors frustrating Taiwan\\'s frontline nurses in their quest to provide good quality care to SARS patients throughout the caring process are first identified herein. It appears that the SARS epidemic highlighted many significant problems not only in contemporary health care delivery systems but also in the interdisciplinary and international health care environments. The value of reciprocity does not only rely on individuals or paperwork: It also needs to be brought about by the support and protection by health institutions of their health care providers (Singer et al., 2003) . Health care providers are expected to deliver services in an atmosphere of utmost integrity that will only occur when providers have courage, feel safe, and are supported to do the right thing in difficult situations (Kerfoot, 1999) . As such, government and health institutes are expected to provide the following forms of help to nurses: (a) offering adequate personal protective material and equipment at all relevant sites; (b) fostering a reciprocal supportive relationship between health team members; (c) offering an accessible counseling \"hotline\" and anti-SARS program to provide timely mutual support and training during quarantine; and (d) providing plausible tangible reimbursement for staff who take extraordinary responsibilities during epidemics and disasters.'], 'disease_ents': ['SARS', 'SARS', 'SARS'], 'chemical_ents': [], 'genetic_ents': []}\n",
      "1200 paragraphs annotated\n",
      "1300 docs evaluated\n",
      "{'id': '0025392e983e096e31076d731275a459', 'section_s': 'analyses', 'article_id_s': '03aa0556aa6d712feb40fd557a3c0fd98338beda', 'size_i': 614, 'name_s': 'analyses', 'text_t': ['Linear regression to determine if the level of total protein influenced the severity of disease (as measured by the duration of treatment) was done twice: First using all calves treated for a specific disease condition and second, using only those that survived the disease. We had postulated that two separate models would evolve. Calves that die, do so early in the course of the disease and thus, number of days treated would not be a good measure of severity in those calves. In fact, the parameter estimates in the two models were nearly identical and so, only the ones using all calves treated are presented.'], 'disease_ents': [], 'chemical_ents': [], 'genetic_ents': []}\n",
      "1300 paragraphs annotated\n",
      "1400 docs evaluated\n",
      "{'id': '002814d8095fdc9d88293b0a96b4112b', 'section_s': 'the e-health literature', 'article_id_s': 'e753a660dac2e8d7f8605f925240cc68fbcba43d', 'size_i': 286, 'name_s': 'the e-health literature', 'text_t': ['Though focused on the US, the existing e-health literature generates key themes for an analysis of progress in other parts of the world, including the East Asian tigers. However, before exploring those themes, we first present some basic contextual information about our five societies.'], 'disease_ents': [], 'chemical_ents': [], 'genetic_ents': []}\n",
      "1400 paragraphs annotated\n",
      "1500 docs evaluated\n",
      "{'id': '002b46ecd3852f799dba68cef413c6b6', 'section_s': 'abstract', 'article_id_s': '22a4d26bde67d878869443ea26140f3d8239bed0', 'size_i': 354, 'name_s': 'abstract', 'text_t': ['using Drosophila melanogaster and Drosophila C Virus as a host-pathogen model 16 system. We found that infected lifespan, viral growth, virus shedding, and viral load at 17 death were all significantly influenced by fly genetic background, sex and female mating 18 status. To understand how variation in each of these traits may generate heterogeneity 19'], 'disease_ents': [], 'chemical_ents': [], 'genetic_ents': []}\n",
      "1500 paragraphs annotated\n",
      "1600 docs evaluated\n",
      "{'id': '002dddcf522e47fabdc9d2f6ab4ecf5d', 'section_s': 'introduction', 'article_id_s': 'c319654afc6b5f346e3d5d05f64dc04c074482fd', 'size_i': 266, 'name_s': 'introduction', 'text_t': ['Manual surveillance is resource-intensive, and can be applied to only a limited patient population. Therefore, automated or semiautomated surveillance algorithms based on electronically available patient data represent a promising alternative to manual surveillance.'], 'disease_ents': [], 'chemical_ents': [], 'genetic_ents': []}\n",
      "1600 paragraphs annotated\n",
      "1700 docs evaluated\n",
      "{'id': '00314587ed5f8000d0c87a7caae14707', 'section_s': 'introduction', 'article_id_s': '2bc2fe97aaf88704800806bac58b85cfdc39a6e2', 'size_i': 511, 'name_s': 'introduction', 'text_t': ['The purpose of this study was to investigate the presence of CoVs in healthy Danish bat populations by: (a) active targeted surveillance of bats associated with the hibernacula at Mønsted; and (b) general surveillance by opportunistic sampling around the country from bats in breeding colonies, day roosts, or caught when flying in the field. In addition, we wanted to relate the different viruses found to the various bat species and to make comparisons to the CoV strains reported from neighbouring countries.'], 'disease_ents': [], 'chemical_ents': [], 'genetic_ents': []}\n",
      "1700 paragraphs annotated\n",
      "1800 docs evaluated\n",
      "{'id': '0034caefd4ac81688641a2273a27f1b0', 'section_s': 'the economic model of anticipatory behavior', 'article_id_s': 'e784769ed85685121bda82e89183d0b756686281', 'size_i': 449, 'name_s': 'the economic model of anticipatory behavior', 'text_t': ['The efficiency of the MRP was quantified by an index defined as the proportional increase in the time-weighted number of uninfected nodes associated with a policy, compared with no policy. This represents the proportional increase in the area above the curve of accumulated infected nodes due to the policy (S1 Fig). The index ranges from 0 to infinity, where 0 is associated with a policy with no benefits in terms of slowing the disease spreading.'], 'disease_ents': [], 'chemical_ents': [], 'genetic_ents': []}\n",
      "1800 paragraphs annotated\n",
      "1900 docs evaluated\n",
      "{'id': '0037658d8e31399f68c9de67607286e1', 'section_s': 'proteomics', 'article_id_s': 'e4fd1ddb11f258d34085b69f59c574b40ded063b', 'size_i': 1103, 'name_s': 'proteomics', 'text_t': ['In an attempt to identify biomarkers and to develop screening tools with high sensitivity and specificity, proteomics technology has been applied to analyze biofluids such as serum, saliva, or urine. A commonly used technique is a surface enhanced laser desorption/ionization time-of-flight (SELDI-TOF), which is a proteomics technique that allows the identification of large numbers of proteins in a short period of time. This technique provides a specific mass spectral profile from each analyzed protein sample. By comparing profiles from affected individuals to those derived from healthy controls, a specific protein signature is obtained (Coombes et al., 2005; Hodgetts et al., 2007) . Agranoff and colleagues have used it to characterise distinct profiles for several microbial infections and to investigate serum responsiveness to M. tuberculosis identifying serum biomarkers from patients with advanced tuberculosis. SELDI-TOF has the potential to identify tuberculosis at an early stage, assisting early diagnosis and therapy, which is important for favorable outcome (Agranoff et al., 2005) .'], 'disease_ents': ['microbial infections', 'tuberculosis', 'tuberculosis'], 'chemical_ents': [], 'genetic_ents': []}\n",
      "1900 paragraphs annotated\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-55-dbf4b6985e19>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     27\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0;34m'text_t'\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     28\u001b[0m                 \u001b[0mparagraph\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'text_t'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'text_t'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 29\u001b[0;31m                 \u001b[0mdoc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnlp\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mparagraph\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'text_t'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     30\u001b[0m                 \u001b[0mparagraph\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'disease_ents'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtext\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mf\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mdoc\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0ments\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlabel_\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'DISEASE'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     31\u001b[0m                 \u001b[0mparagraph\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'chemical_ents'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtext\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mf\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mdoc\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0ments\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlabel_\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'CHEMICAL'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/spacy/language.py\u001b[0m in \u001b[0;36m__call__\u001b[0;34m(self, text, disable, component_cfg)\u001b[0m\n\u001b[1;32m    988\u001b[0m                 \u001b[0merror_handler\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mproc\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_error_handler\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    989\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 990\u001b[0;31m                 \u001b[0mdoc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mproc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdoc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mcomponent_cfg\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    991\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    992\u001b[0m                 \u001b[0;31m# This typically happens if a component is not initialized\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Master/TFM/bio-nlp/NER_processing.py\u001b[0m in \u001b[0;36m__call__\u001b[0;34m(self, doc)\u001b[0m\n\u001b[1;32m     57\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__call__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdoc\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mDoc\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m->\u001b[0m \u001b[0mDoc\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     58\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mentities\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mclass_entities\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mEntities\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdoc\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 59\u001b[0;31m         \u001b[0mprocess_by_paragraph\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdoc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mentities\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     60\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mentities\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpostprocessing\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     61\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mdoc\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Master/TFM/bio-nlp/NER_processing.py\u001b[0m in \u001b[0;36mprocess_by_paragraph\u001b[0;34m(doc, entities)\u001b[0m\n\u001b[1;32m     31\u001b[0m         \u001b[0mentities\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend_new_entities\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mchemical_results\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     32\u001b[0m         \u001b[0mgenetic_service\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msentence_to_process\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mparagraph\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 33\u001b[0;31m         \u001b[0mgenetic_results\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mgenetic_service\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpredict\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     34\u001b[0m         \u001b[0mentities\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend_new_entities\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgenetic_results\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     35\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Master/TFM/bio-nlp/bioprocessor.py\u001b[0m in \u001b[0;36mpredict\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m     36\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     37\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mpredict\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 38\u001b[0;31m         \u001b[0mresults\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpipeline\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msequence\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     39\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moffset\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     40\u001b[0m             \u001b[0;32mfor\u001b[0m \u001b[0mresult\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mresults\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/transformers/pipelines/token_classification.py\u001b[0m in \u001b[0;36m__call__\u001b[0;34m(self, inputs, **kwargs)\u001b[0m\n\u001b[1;32m    174\u001b[0m                     \u001b[0;32mwith\u001b[0m \u001b[0mtorch\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mno_grad\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    175\u001b[0m                         \u001b[0mtokens\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mensure_tensor_on_device\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m**\u001b[0m\u001b[0mtokens\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 176\u001b[0;31m                         \u001b[0mentities\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmodel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m**\u001b[0m\u001b[0mtokens\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcpu\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnumpy\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    177\u001b[0m                         \u001b[0minput_ids\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtokens\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"input_ids\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcpu\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnumpy\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    178\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_call_impl\u001b[0;34m(self, *input, **kwargs)\u001b[0m\n\u001b[1;32m    887\u001b[0m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_slow_forward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    888\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 889\u001b[0;31m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mforward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    890\u001b[0m         for hook in itertools.chain(\n\u001b[1;32m    891\u001b[0m                 \u001b[0m_global_forward_hooks\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/transformers/models/bert/modeling_bert.py\u001b[0m in \u001b[0;36mforward\u001b[0;34m(self, input_ids, attention_mask, token_type_ids, position_ids, head_mask, inputs_embeds, labels, output_attentions, output_hidden_states, return_dict)\u001b[0m\n\u001b[1;32m   1677\u001b[0m         \u001b[0mreturn_dict\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mreturn_dict\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0mreturn_dict\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m \u001b[0;32melse\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mconfig\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0muse_return_dict\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1678\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1679\u001b[0;31m         outputs = self.bert(\n\u001b[0m\u001b[1;32m   1680\u001b[0m             \u001b[0minput_ids\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1681\u001b[0m             \u001b[0mattention_mask\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mattention_mask\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_call_impl\u001b[0;34m(self, *input, **kwargs)\u001b[0m\n\u001b[1;32m    887\u001b[0m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_slow_forward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    888\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 889\u001b[0;31m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mforward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    890\u001b[0m         for hook in itertools.chain(\n\u001b[1;32m    891\u001b[0m                 \u001b[0m_global_forward_hooks\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/transformers/models/bert/modeling_bert.py\u001b[0m in \u001b[0;36mforward\u001b[0;34m(self, input_ids, attention_mask, token_type_ids, position_ids, head_mask, inputs_embeds, encoder_hidden_states, encoder_attention_mask, past_key_values, use_cache, output_attentions, output_hidden_states, return_dict)\u001b[0m\n\u001b[1;32m    969\u001b[0m             \u001b[0mpast_key_values_length\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mpast_key_values_length\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    970\u001b[0m         )\n\u001b[0;32m--> 971\u001b[0;31m         encoder_outputs = self.encoder(\n\u001b[0m\u001b[1;32m    972\u001b[0m             \u001b[0membedding_output\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    973\u001b[0m             \u001b[0mattention_mask\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mextended_attention_mask\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_call_impl\u001b[0;34m(self, *input, **kwargs)\u001b[0m\n\u001b[1;32m    887\u001b[0m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_slow_forward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    888\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 889\u001b[0;31m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mforward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    890\u001b[0m         for hook in itertools.chain(\n\u001b[1;32m    891\u001b[0m                 \u001b[0m_global_forward_hooks\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/transformers/models/bert/modeling_bert.py\u001b[0m in \u001b[0;36mforward\u001b[0;34m(self, hidden_states, attention_mask, head_mask, encoder_hidden_states, encoder_attention_mask, past_key_values, use_cache, output_attentions, output_hidden_states, return_dict)\u001b[0m\n\u001b[1;32m    566\u001b[0m                 )\n\u001b[1;32m    567\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 568\u001b[0;31m                 layer_outputs = layer_module(\n\u001b[0m\u001b[1;32m    569\u001b[0m                     \u001b[0mhidden_states\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    570\u001b[0m                     \u001b[0mattention_mask\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_call_impl\u001b[0;34m(self, *input, **kwargs)\u001b[0m\n\u001b[1;32m    887\u001b[0m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_slow_forward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    888\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 889\u001b[0;31m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mforward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    890\u001b[0m         for hook in itertools.chain(\n\u001b[1;32m    891\u001b[0m                 \u001b[0m_global_forward_hooks\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/transformers/models/bert/modeling_bert.py\u001b[0m in \u001b[0;36mforward\u001b[0;34m(self, hidden_states, attention_mask, head_mask, encoder_hidden_states, encoder_attention_mask, past_key_value, output_attentions)\u001b[0m\n\u001b[1;32m    454\u001b[0m         \u001b[0;31m# decoder uni-directional self-attention cached key/values tuple is at positions 1,2\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    455\u001b[0m         \u001b[0mself_attn_past_key_value\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpast_key_value\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;36m2\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0mpast_key_value\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m \u001b[0;32melse\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 456\u001b[0;31m         self_attention_outputs = self.attention(\n\u001b[0m\u001b[1;32m    457\u001b[0m             \u001b[0mhidden_states\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    458\u001b[0m             \u001b[0mattention_mask\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_call_impl\u001b[0;34m(self, *input, **kwargs)\u001b[0m\n\u001b[1;32m    887\u001b[0m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_slow_forward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    888\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 889\u001b[0;31m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mforward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    890\u001b[0m         for hook in itertools.chain(\n\u001b[1;32m    891\u001b[0m                 \u001b[0m_global_forward_hooks\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/transformers/models/bert/modeling_bert.py\u001b[0m in \u001b[0;36mforward\u001b[0;34m(self, hidden_states, attention_mask, head_mask, encoder_hidden_states, encoder_attention_mask, past_key_value, output_attentions)\u001b[0m\n\u001b[1;32m    385\u001b[0m         \u001b[0moutput_attentions\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    386\u001b[0m     ):\n\u001b[0;32m--> 387\u001b[0;31m         self_outputs = self.self(\n\u001b[0m\u001b[1;32m    388\u001b[0m             \u001b[0mhidden_states\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    389\u001b[0m             \u001b[0mattention_mask\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_call_impl\u001b[0;34m(self, *input, **kwargs)\u001b[0m\n\u001b[1;32m    887\u001b[0m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_slow_forward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    888\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 889\u001b[0;31m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mforward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    890\u001b[0m         for hook in itertools.chain(\n\u001b[1;32m    891\u001b[0m                 \u001b[0m_global_forward_hooks\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/transformers/models/bert/modeling_bert.py\u001b[0m in \u001b[0;36mforward\u001b[0;34m(self, hidden_states, attention_mask, head_mask, encoder_hidden_states, encoder_attention_mask, past_key_value, output_attentions)\u001b[0m\n\u001b[1;32m    251\u001b[0m         \u001b[0moutput_attentions\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    252\u001b[0m     ):\n\u001b[0;32m--> 253\u001b[0;31m         \u001b[0mmixed_query_layer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mquery\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhidden_states\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    254\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    255\u001b[0m         \u001b[0;31m# If this is instantiated as a cross-attention module, the keys\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_call_impl\u001b[0;34m(self, *input, **kwargs)\u001b[0m\n\u001b[1;32m    887\u001b[0m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_slow_forward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    888\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 889\u001b[0;31m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mforward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    890\u001b[0m         for hook in itertools.chain(\n\u001b[1;32m    891\u001b[0m                 \u001b[0m_global_forward_hooks\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/torch/nn/modules/linear.py\u001b[0m in \u001b[0;36mforward\u001b[0;34m(self, input)\u001b[0m\n\u001b[1;32m     92\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     93\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mforward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0minput\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mTensor\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m->\u001b[0m \u001b[0mTensor\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 94\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mF\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlinear\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mweight\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbias\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     95\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     96\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mextra_repr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m->\u001b[0m \u001b[0mstr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/torch/nn/functional.py\u001b[0m in \u001b[0;36mlinear\u001b[0;34m(input, weight, bias)\u001b[0m\n\u001b[1;32m   1751\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mhas_torch_function_variadic\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mweight\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1752\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mhandle_torch_function\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlinear\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mweight\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mweight\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbias\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mbias\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1753\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mtorch\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_C\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_nn\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlinear\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mweight\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbias\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1754\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1755\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "paragraphs_processed = []\n",
    "print(\"reading from solr..\")\n",
    "counter = 0\n",
    "completed = False\n",
    "window_size= 100\n",
    "cursor = \"*\"\n",
    "while (not completed):\n",
    "    old_counter = counter        \n",
    "    \n",
    "    try:\n",
    "        paragraphs = solr_local.search(q=\"*:*\", rows=window_size, start=counter, sort=\"id asc\")\n",
    "#         cursor = paragraphs.nextCursorMark\n",
    "        \n",
    "        for p in paragraphs:\n",
    "            paragraph = {}\n",
    "            paragraph['id'] = p['id']\n",
    "            if ('section_s' in p):\n",
    "                paragraph['section_s'] = p['section_s']\n",
    "            if ('id' in p):\n",
    "                paragraph['id'] = p['id']\n",
    "            if ('article_id_s' in p):\n",
    "                paragraph['article_id_s'] = p['article_id_s']\n",
    "            if ('size_i' in p):\n",
    "                paragraph['size_i'] = p['size_i']\n",
    "            if ('name_s' in p):\n",
    "                paragraph['name_s'] = p['name_s']\n",
    "            if ('text_t' in p):\n",
    "                paragraph['text_t'] = p['text_t']\n",
    "                doc = nlp(str(paragraph['text_t']))            \n",
    "                paragraph['disease_ents'] = [f.text for f in doc.ents if f.label_ == 'DISEASE']\n",
    "                paragraph['chemical_ents'] = [f.text for f in doc.ents if f.label_ == 'CHEMICAL']\n",
    "                paragraph['genetic_ents'] = [f.text for f in doc.ents if f.label_ == 'GENETIC']\n",
    "            paragraphs_processed.append(paragraph)\n",
    "            \n",
    "        counter += len(paragraphs)\n",
    "        print(counter,\"docs evaluated\")\n",
    "        \n",
    "        if counter % window_size == 0:\n",
    "            print(paragraphs_processed[0])\n",
    "            solr_local.add(paragraphs_processed)\n",
    "            print(counter, 'paragraphs annotated')\n",
    "            paragraphs_processed=[]\n",
    "            \n",
    "        if (old_counter == counter):\n",
    "            print(\"done!\")\n",
    "            break\n",
    "            \n",
    "    except Exception as e:\n",
    "        print(repr(e))\n",
    "        print(\"Solr query error. Wait for 5secs..\")\n",
    "        time.sleep(5.0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4094d3af-b071-44d2-bb2c-567110b95894",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "649f2539-c4cb-4718-be3f-1c5382360e8e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "db1533c1-740c-45bf-aa58-a2ac7254b280",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c3076e1b-2f11-4c93-8eb6-851ef5d40286",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "a8db1c3e-fc2a-410f-a23c-fc4b3801ede4",
   "metadata": {},
   "outputs": [],
   "source": [
    "class CORD19Processor:\n",
    "    def __init__(self,window_size):\n",
    "        self.covid_articles = pysolr.Solr('http://librairy.linkeddata.es/data/covid', timeout=10)\n",
    "        self.covid_paragraphs = pysolr.Solr('http://librairy.linkeddata.es/data/covid-paragraphs', timeout=10)\n",
    "        self.window_size = window_size\n",
    "        self.annotated_paragraphs = self.annotate_paragraphs()\n",
    "    \n",
    "    \n",
    "    def annotate_paragraphs(self):\n",
    "        counter = 0\n",
    "        completed = False\n",
    "        cursor = \"*\"\n",
    "        i=0\n",
    "        j=0\n",
    "        processed_paragraphs = []\n",
    "        while (not completed):\n",
    "            old_counter = counter\n",
    "            try:\n",
    "                results = self.covid_paragraphs.search(q=\"*:*\",cursorMark=cursor, rows=self.window_size,sort=\"id asc\")\n",
    "                cursor = results.nextCursorMark\n",
    "                counter += len(results)\n",
    "                paragraphs = []\n",
    "                print(len(results))\n",
    "                for result in results:\n",
    "                    paragraph = {}\n",
    "                    if ('section_s' in result):\n",
    "                        paragraph['section_s'] = result['section_s']\n",
    "                    if ('id' in result):\n",
    "                        paragraph['id'] = result['id']\n",
    "                    if ('article_id_s' in result):\n",
    "                        paragraph['article_id_s'] = result['article_id_s']\n",
    "                    if ('text_t' in result):\n",
    "                        paragraph['text_t'] = result['text_t']\n",
    "                    if ('size_i' in result):\n",
    "                        paragraph['size_i'] = result['size_i']\n",
    "                    if ('name_s' in result):\n",
    "                        paragraph['name_s'] = result['name_s']\n",
    "                    paragraphs.append(paragraph)\n",
    "                    j+=1\n",
    "                    if j%100000 == 0:\n",
    "                        print(j, 'paragraphs retrieved')\n",
    "                print(len(paragraphs),'paragraphs retrieved')\n",
    "                if (old_counter == counter):\n",
    "                    print('Final')\n",
    "                    break\n",
    "        \n",
    "                for paragraph in paragraphs:\n",
    "                    doc = nlp(paragraph['text_t'])\n",
    "                    paragraph['disease_ents'] = [f for f in doc.ents if f.label_ == 'DISEASE']\n",
    "                    paragraph['chemical_ents'] = [f for f in doc.ents if f.label_ == 'CHEMICAL']\n",
    "                    paragraph['genetic_ents'] = [f for f in doc.ents if f.label_ == 'GENETIC']\n",
    "                    i+=1\n",
    "                    if i%10000 == 0:\n",
    "                        print(i, 'paragraphs processed')\n",
    "                    processed_paragraphs.append(paragraph)\n",
    "               \n",
    "                completed = True\n",
    "            except Exception as e:\n",
    "                print(repr(e))\n",
    "                        \n",
    "        return paragraph\n",
    "\n",
    "        \n",
    "            \n",
    "            \n",
    "            \n",
    "            "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "35110d5d-4cf3-4762-abc3-eb82c6b35966",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2103891\n",
      "100000 paragraphs retrieved\n",
      "200000 paragraphs retrieved\n",
      "300000 paragraphs retrieved\n",
      "400000 paragraphs retrieved\n",
      "500000 paragraphs retrieved\n",
      "600000 paragraphs retrieved\n",
      "700000 paragraphs retrieved\n",
      "800000 paragraphs retrieved\n",
      "900000 paragraphs retrieved\n",
      "1000000 paragraphs retrieved\n",
      "1100000 paragraphs retrieved\n",
      "1200000 paragraphs retrieved\n",
      "1300000 paragraphs retrieved\n",
      "1400000 paragraphs retrieved\n",
      "1500000 paragraphs retrieved\n",
      "1600000 paragraphs retrieved\n",
      "1700000 paragraphs retrieved\n",
      "1800000 paragraphs retrieved\n",
      "1900000 paragraphs retrieved\n",
      "2000000 paragraphs retrieved\n",
      "2100000 paragraphs retrieved\n",
      "2103891 paragraphs retrieved\n",
      "100 paragraphs processed\n",
      "{'section_s': 'background', 'id': '0002d368d74a23c546db9bc6098c37fe', 'article_id_s': '744f2ba295872a8f27eae0204ba16b3afb3bd7c9', 'text_t': 'Aseptic meningitis, encephalitis, and myelitis are inflammatory conditions of the central nervous system (CNS) (meninges, brain, and spinal cord, respectively). Disease is caused by a variety of infectious pathogens, but viruses cause most disease. Viruses gain entry to the CNS via the bloodstream (enteroviruses and arboviruses) or by direct neuronal spread (HSV and rabies). Pathogenesis may involve direct viral invasion or a vigorous virus-specific immune response resulting in damage to the neurons and supporting cells. Alternatively, infection may trigger activation of an immune response specific for the oligodendroglia or the myelin components themselves. In the latter case, disease may follow an upper respiratory tract or other infection and primarily take the form of a demyelinating process. This disease is commonly termed postinfectious encephalomyelitis or acute disseminated encephalomyelitis.', 'size_i': 913, 'name_s': 'background', 'disease_ents': [Aseptic meningitis, encephalitis, myelitis, inflammatory conditions, rabies, upper respiratory tract, demyelinating process, postinfectious encephalomyelitis, acute disseminated encephalomyelitis], 'chemical_ents': [], 'genetic_ents': []}\n",
      "200 paragraphs processed\n",
      "{'section_s': 'conclusion', 'id': '0005f9e08f2de1e27d168dea5b42af2c', 'article_id_s': '2a066d13074f224d0078ffe3abbbc042a5a1883b', 'text_t': 'Growing number of atomic structures in PDB facilitate the study of RBP-RNA interactions. Diversity of structural rearrangement of RNA induces the conformational transition to its partner RBP. Mutual folding of RNA and protein in ribonucleoprotein complexes is a common phenomenon and is probably ubiquitous. A recent analysis suggested that the IDPs are proteins waiting for their partner to bind and acquire a particular conformation [96] . Presently, a significant number of atomic structures of RBPs in bound and unbound conformations is available in the PDB [97] . These structural information shows that polymorphic conformation is intrinsic to RBP. In this study, we propose that polymorphic conformation of RBPs promote structural flexibility, which considerably influences the conformational dynamics of RBP-RNA interaction. Indeed, RNA recognition by RBPs is more than a simple handshake and is associated with a variety of conformational flexibility in both the partners. Hence, the lack of structural integrity, camouflaged in conformational polymorphism, is highly favoured in this kind of molecular recognition. Although, the PDB covers only about 15 % of the known protein-RNA complexes of human RBP, many more structural information is required in near future to make a complete repertoire of RBP-RNA interactions, which eventually can lead us to a better understanding of human diseases associated with this recognition process.', 'size_i': 1444, 'name_s': 'conclusion', 'disease_ents': [], 'chemical_ents': [], 'genetic_ents': [RBP-RNA, RBP, IDPs, RBPs, RBP, RBPs, RBP-RNA, human RBP, RBP-RNA]}\n",
      "300 paragraphs processed\n",
      "{'section_s': 'discussion', 'id': '0008cf6c8ab1d58af9f4389dfa2b2808', 'article_id_s': 'PMC5016003', 'text_t': 'The papers included in this review are in the majority originated in previous decades,\\nwhich demonstrates the need of current research conduction because there were advances\\nin the technology of the equipment used in CSD, for example, in ultrasonic cleaner. The\\nultrasonic cleaner complement the manual cleaning or are self-cleaning in cleaning the\\nMD with simple conformation depending on the power of ultrasound. They are mainly\\nsuitable for complementary cleaning of the complex forming MD and operate on the\\nprinciple of cavitation (sound waves propagated in an aqueous medium fragmenting rupture\\nor dirt adhered to products)\\n7\\n. Despite using the same principle, the equipment currently available are modern,\\nof different dimensions and efficiency when compared to the washer used in the study\\nincluded in this review\\n14\\n, and the fact that they are usually operated capped. A limitation of this study\\nwas the definition for the recovery of only one microorganism (Pseudomonas\\naeruginosa) and it was rated as partially conclusive.', 'size_i': 1035, 'name_s': 'discussion', 'disease_ents': [], 'chemical_ents': [], 'genetic_ents': []}\n",
      "400 paragraphs processed\n",
      "{'section_s': 'introduction', 'id': '000c07825b8984b994ba9d509041455f', 'article_id_s': '141e9bd4760675a75fb1ceba5a0c6c352078c92d', 'text_t': 'The implications of targeted alterations of the reovirus genome are highly significant to current research focused on therapeutic use of reovirus as an oncolytic agent (Kelly et al., 2009) and to conceptualization of reovirus-vectored vaccines. Goals for our continuing work with this technology include performance improvement in the form of increased efficiency of virus recovery, broadened scope of dissectible viral properties and phenotypes, expanded spectrum of non-viral nucleic acids inserted into the viral genome, shortened time to virus recovery, and reduced reagent preparation time and expense. Progress in these key aspects of the rescue system will accelerate basic studies of reovirus biology and development of reovirus for clinical applications.', 'size_i': 763, 'name_s': 'introduction', 'disease_ents': [], 'chemical_ents': [], 'genetic_ents': []}\n",
      "500 paragraphs processed\n",
      "{'section_s': 'emergence of a large-plaque variant of cvb3 following inoculation of mice. ::: results', 'id': '000f90cb43b0ceb19e25fa462d15115a', 'article_id_s': 'PMC4817249', 'text_t': 'To determine what factors contribute to the emergence of the large-plaque variant, we used different inoculation routes and examined the plaque morphology of virus from different tissues of immunocompetent or immunodeficient mice. IFNAR−/− mice and immunocompetent IFNAR+/+ mice were orally, intraperitoneally (i.p.), or intramuscularly (i.m.) inoculated with 5 × 107, 5 × 107, or 2 × 106 PFU of CVB3, respectively. Fecal samples were collected from orally inoculated mice at 72 hpi; livers, hearts, and spleens were harvested from i.p. inoculated mice at 48 hpi; and muscles and livers were harvested from i.m. inoculated mice at 48 hpi. We measured the virus plaque sizes and quantified the percentages of large plaques in different tissues (Fig. 2). First, we found that there were more large-plaque variants in IFNAR+/+ mice than in IFNAR−/− mice. The increased emergence of the large-plaque virus in immunocompetent mice correlates with reduced viral replication, indicating that increased selective pressure may drive the emergence of the large-plaque variant. Second, large-plaque variants were present in feces from orally inoculated mice and visceral tissues from i.p. or i.m. inoculated mice but not in muscle from i.m. inoculated IFNAR−/− mice. We hypothesize that increased selective pressure in the intestine and peritoneal cavity facilitated the emergence of large-plaque variants and/or the large-plaque variant has a fitness advantage in selected tissues. Overall, these results suggest that the large-plaque variant existed at a low level in our inoculum but was enriched following viral replication in several tissues of infected mice.', 'size_i': 1653, 'name_s': 'emergence of a large-plaque variant of cvb3 following inoculation of mice. ::: results', 'disease_ents': [], 'chemical_ents': [], 'genetic_ents': []}\n",
      "600 paragraphs processed\n",
      "{'section_s': 'malades atteints du sida', 'id': '0012c43b20df8849c0054c0d780b819f', 'article_id_s': 'd28fe6771927afbbcaad48684cba1e79d2e5312a', 'text_t': \"Le sarcome de Kaposi, sans être totalement spécifi que du sida, y est malgré tout très lié par sa grande fréquence au cours de cet état d'immunodépression, essentiellement chez les hommes homosexuels. L'atteinte est essentiellement cutanée mais aussi trachéo-bronchique et pulmonaire. L'aspect radiologique de l'atteinte pulmonaire est maintenant bien connu, fait de nodules ou de masses à contours spiculés, volontiers entourés de verre dépoli, d' épaississements péribronchovasculaires et d' épanchements pleuraux bilatéraux à un stade tardif, les vraies adénopathies y étant en fait assez rares [41] . Cette pathologie semble sous-tendue par une infection à Herpes virus type HHV-8, qu'elle soit liée ou non au VIH, de même que la maladie de Castelman disséminée et le lymphome des séreuses [53] . Les chimiothérapies avaient permis une amélioration passagère de l'état des patients sans en améliorer le pronostic [14] , mais là aussi, c'est l'effi cacité du traitement antirétroviral qui a permis une amélioration de ce pronostic et une régression drastique de cette pathologie [39, 42] .\", 'size_i': 1092, 'name_s': 'malades atteints du sida', 'disease_ents': [adénopathies, virus], 'chemical_ents': [], 'genetic_ents': []}\n",
      "700 paragraphs processed\n",
      "{'section_s': 'marmosets', 'id': '001648769bcdbc05420c83764a381bd5', 'article_id_s': '30b4b3db4476c5df7a3797c56537a46b4db61eac', 'text_t': 'To clarify the clinically relevant effect of antiviral therapy against MERS-CoV infection, the common marmosets were initially intratracheally infected with 5 × 10 6 TCID 50 of MERS-CoV, followed by intravenous inoculation with MCA1 at 2 (G3) or 12 (G4) hours, at 20 mg/kg. The efficacy of the treatment was determined. At 72 hours after infection, the mean clinical score for the G3 group was 2.7, compared with 6.3 for the G4 group, and both lower scores than for the M group ( Figure 3A) .', 'size_i': 492, 'name_s': 'marmosets', 'disease_ents': [MERS-CoV infection, MERS-CoV], 'chemical_ents': [], 'genetic_ents': []}\n",
      "800 paragraphs processed\n",
      "{'section_s': 'roles of infectious disease-related laws', 'id': '00196bcb302865ebd86538ca22dc6a4a', 'article_id_s': '9d689209d9886db326dc65759ef929ac56832d93', 'text_t': 'Since the 2000s, international organizations such as the World Health Organization (WHO), the International Food and Agriculture Organization, and the World Organization for Animal Health (Office International des epizooties) have actively promoted international collaboration to respond to uncertainty and mutation potential of pathogens and to prevent spread of emerging infectious diseases. These are also closely related with the reason national health care systems made efforts to establish better infrastructure. These international organizations announced an international common strategy for control of avian influenza through the Beijing Declaration in 2006, by which they asked for development of an integrated action plan for each country and practical performances [5] .', 'size_i': 782, 'name_s': 'roles of infectious disease-related laws', 'disease_ents': [avian influenza], 'chemical_ents': [], 'genetic_ents': []}\n",
      "900 paragraphs processed\n",
      "{'section_s': 'regulation of ifn signaling pathway', 'id': '001c50512fd434ad6ab4d64096cdccbe', 'article_id_s': '7334f0e1c2973f27bed7c437ed9849d7b494686d', 'text_t': 'Minimal IFN production is noticeable under arterivirus infection, but PRRs still senses invading arteriviruses. An increased activation of IFN-␤ promoter has been noticed in PRRSV-infected PAMs and MARC-145 cells, and TLR3 seems to be involved (Miller et al., 2004 (Miller et al., , 2009 Shi et al., 2010) . TLR7 is also thought to be essential for IFN production in LDV-and PRRSV-stimulated IFN response in pDCs (Ammann et al., 2009; Baumann et al., 2013) . A study using EAV in specific gene-knockout mouse embryonic fibroblasts suggests that both MDA-5 and RIG-I play a role in counteracting viral infection (van Kasteren et al., 2013) . In PRRSV-infected PAMs and DCs, expression of TLR3 and TLR7 is inhibited at early infection but restored later (Chaung et al., 2010) . The TLR3 and TLR7 Inhibits production of type I IFNs and impairs IFN promoter activity Han et al. (2014) expressions are delayed in PRRSV-infected tracheobronchial lymph nodes (Liu et al., 2009 ), but no difference is observed in PAMs among different isolates of PRRSV for transcription of TLR3, TLR7, and TLR9 (Kuzemtseva et al., 2014) .', 'size_i': 1114, 'name_s': 'regulation of ifn signaling pathway', 'disease_ents': [arterivirus], 'chemical_ents': [], 'genetic_ents': [IFN-␤ promoter, TLR3, TLR7, MDA-5, RIG-I, TLR3, TLR7, TLR3, TLR7, type I IFNs, IFN promoter, TLR3, TLR7, TLR9]}\n",
      "1000 paragraphs processed\n",
      "{'section_s': 'clinical features', 'id': '001f0dd5330e2c8c879139ba0850c60e', 'article_id_s': 'PMC7123916', 'text_t': 'After an incubation period of 10–14 days, prodromal measles begins with fever of 39–40 C, cough, coryza, and conjunctivitis (Fig. 8.2). The initial presentation resembles a cold except that fever is an early sign. These symptoms worsen over a 2- to 4-day period; sneezing, rhinitis, and congestion are common. The cough is frequently troublesome and often has a brassy quality, suggesting laryngeal and tracheal involvement. If the mucous membranes lining the cheeks opposite the molar teeth are examined, there will be found in the majority of patients a few white to bluish-gray spots the size of a pinhead with red margins. These are known as Koplik’s spots and are a pathognomonic exanthem of measles.\\n\\n', 'size_i': 707, 'name_s': 'clinical features', 'disease_ents': [measles, fever, cough, coryza, conjunctivitis, fever, rhinitis, congestion, Koplik’s spots, measles], 'chemical_ents': [], 'genetic_ents': []}\n",
      "1100 paragraphs processed\n",
      "{'section_s': '| sample collection', 'id': '0022693d3dd20ff58bd6b82eeba44825', 'article_id_s': '8b264d2d749532dff506b14356b606029bb94035', 'text_t': 'Environmental swabs were taken from three discrete areas of markets to determine the level of influenza virus A contamination of LBM environments. These discrete areas represented different poultry-related work activities which had previously been recommended to be included in routine monitoring and surveillance programmes for avian influenza viruses in LBMs (Indriani et al., 2010) :', 'size_i': 386, 'name_s': '| sample collection', 'disease_ents': [influenza], 'chemical_ents': [], 'genetic_ents': []}\n",
      "1200 paragraphs processed\n",
      "{'section_s': 'results', 'id': '00252fb71133f6ccf131a420201932c5', 'article_id_s': '9146df7c1faca31203b1e903f59662067faebb69', 'text_t': 'All serum samples were screened in triplicate at a dilution of 1:20. A specimen was considered as screening positive if the average of the three wells containing the sera exhibited a >50% reduction in luminescence in comparison to the luminescence negative (BSA) or no serum control. Thirteen sera (9 animal exposed and 4 controls) screened positive at 1:20, but upon titration of these sera in subsequent assays, none of them were reproducibly able to neutralize the MERS-CoV pseudotyped lentiviral vector at a serum dilution of 1:20.', 'size_i': 535, 'name_s': 'results', 'disease_ents': [MERS-CoV], 'chemical_ents': [], 'genetic_ents': []}\n",
      "1300 paragraphs processed\n",
      "{'section_s': 'quantitative analysis', 'id': '002810ef95fd03f07cb57c61947da949', 'article_id_s': '356f0fd788357570d8316ea566c88002554745b0', 'text_t': 'The in vitro cleavage efficiency of K48 and K63-linked polyubiquitin chains was expressed as the ratio of the amount of Ub plus Ub2 over total Ub (Ub+Ub2+Ub3+Ub4+Ub5+Ub6+Ub7) by measuring the band density in each lance of SDS-PAGE images. The ratio of Ub2 over Ub1 was calculated by band density of Ub2 over Ub1). The relative DUB activity of PLP2 mutants was measured by the corresponding band density of polyubiquitin conjugates and then normalized against actin and expression level of PLP2. All the data are shown as mean± SEM with 3 independent experiments. ', 'size_i': 563, 'name_s': 'quantitative analysis', 'disease_ents': [], 'chemical_ents': [], 'genetic_ents': [K48, K63-linked polyubiquitin chains, Ub, Ub2, Ub, Ub2, Ub1, Ub2, Ub1, DUB, PLP2 mutants, polyubiquitin, actin, PLP2]}\n",
      "1400 paragraphs processed\n",
      "{'section_s': 'expert commentary and five year view on personalized vaccinology', 'id': '002b2920d49528625d58bb43ab7f3db2', 'article_id_s': 'e1695555871060339e21718b9e515d043be3a9ca', 'text_t': 'Vaccines have made remarkable contributions to human health and remain one of the most cost-effective medical interventions yet devised. [2] However, there are major diseases of global impact for which vaccines do not exist, but are ideal targets for the next generation of vaccine products. [174] Historically, vaccines have been made by empirical methods and are applied to entire populations in a one size fits all approach. While this approach has been tremendously successful, it has a number of disadvantages: candidate vaccines are largely developed in animal models and then must enter large and expensive clinical trials to demonstrate safety and efficacy; vaccine development also relies on an inconsistent or incomplete understanding of correlates of protection. Current issues that affect vaccine development, particularly for complex and hypervariable organisms, require a shift away from empirical approaches toward directed vaccine development approaches that capitalize on: cutting edge technologies; current understanding of innate and adaptive immune function; an increased awareness of vaccine safety; ability to predict and reduce adverse events; the effect of host-pathogen interactions; and the influence of the microbiome. Most important is the realization that human diversity (e.g. genetic, epigenetic, chronologic, and environmental) results in a wide spectrum of inter-individual vaccine responses.', 'size_i': 1425, 'name_s': 'expert commentary and five year view on personalized vaccinology', 'disease_ents': [], 'chemical_ents': [], 'genetic_ents': []}\n",
      "1500 paragraphs processed\n",
      "{'section_s': 'birds', 'id': '002dc9d0c131064115ae6ecc03a1d53f', 'article_id_s': '26490b8cf41325ee33f037bb32d289abaea8e4c2', 'text_t': 'In turkeys, AEEC strains were implicated in the pathogenesis of poult enteritis-mortality syndrome (PEMS), characterized by diarrhoea, depression and dehydration (Guy et al., 2000; Pakpinyo et al., 2002) . AE lesions or AEEC strains (lacking genes for VT or BFP), were detected in 10 of 12 affected flocks (Pakpinyo et al., 2002) . Inoculation of young turkeys with putative turkey EPEC produced severe disease when accompanied by turkey coronavirus (TCV), but only symptomless intestinal AE lesions when inoculated alone (Guy et al., 2000; Pakpinyo et al., 2002) . TCV inoculated alone produced mild disease. Several serogroups, including O111, were represented amongst the turkey EPEC strains implicated in PEMS (Pakpinyo et al., 2002) .', 'size_i': 739, 'name_s': 'birds', 'disease_ents': [poult enteritis-mortality syndrome, PEMS, diarrhoea, depression, dehydration, AE lesions, coronavirus, intestinal AE lesions, PEMS], 'chemical_ents': [], 'genetic_ents': []}\n",
      "1600 paragraphs processed\n",
      "{'section_s': 'discussion', 'id': '0031441e691c3ffa26ed0cb12b62e4a4', 'article_id_s': 'PMC3443008', 'text_t': 'It is worth noting that the influence of employment status on monthly and daily patterns of charcoal burning suicides differed considerably between men and women. According to our data, the monthly variations in suicide risk were more pronounced for non-working men and working women, whereas the daily variations were more apparent for working men and non-working women. It appears that economic activity status either strengthens or weakens the magnitude of the temporal variations in charcoal burning suicide, rather than serving as a determinant that alters the temporal distribution. This may indicate that active employment only serves as a stabilizing factor that protects individuals from various life-threatening forms of behavior (e.g. alcohol consumption or misuse of illicit drugs) [39] and thereby implicitly influences the timing of suicide among specific groups of individuals, but not the whole population. Accordingly, the nature and magnitude of these effects should be recognized in future studies to develop a more effective approach to suicide prevention in Hong Kong.', 'size_i': 1089, 'name_s': 'discussion', 'disease_ents': [charcoal burning, charcoal burning, suicide], 'chemical_ents': [alcohol], 'genetic_ents': []}\n",
      "1700 paragraphs processed\n",
      "{'section_s': 'ai ai bi bi', 'id': '0034c4946c7a005d81af96addc044116', 'article_id_s': '021aa59c28ff346dcdd67e70d20b4dc2af4bc3c9', 'text_t': 'Similarly, we have', 'size_i': 18, 'name_s': 'ai ai bi bi', 'disease_ents': [], 'chemical_ents': [], 'genetic_ents': []}\n",
      "1800 paragraphs processed\n",
      "{'section_s': 'body', 'id': '0037608efdd10ba28d6fc6acfd58cce3', 'article_id_s': '1d3f0f0ee215dae40294661ec3db5310ef743ace', 'text_t': 'A search of the published literature revealed the use of two types of continuous wireless thermometry systemsone type that measures core temperature using a disposable ingestible capsule (Byrne & Lim, 2007) and another type that measures skin temperature using a disposable dermal patch (Racinais, Gaoua, & Grantham, 2008) . In these systems, temperature readings are transmitted wirelessly at regular intervals to a patient monitor or data recorder. Ingestible thermometers have been used to study the thermoregulation of sports persons and soldiers during physical exercise (Gant, Atkinson, & Williams, 2006; Lim, Byrne, & Lee, 2008) and investigated for use in monitoring body temperature during cardiac surgery (Markides, Omorphos, Kotoulas, & Prendergast, 2007) , but they are not suitable for routine monitoring of temperature in young children. The published literature also revealed the use of a skin-contact thermometer that automatically measures skin temperature at predetermined intervals, but its temperature readings are not transmitted wirelessly, are stored in the thermometer, and have to be downloaded to a computer by wired means (Sarabia, Rol, Mendiola, & Madrid, 2008; van Marken Lichtenbelt et al., 2006) . In comparison, although the ThermoSENSOR is also a skin-contact thermometer, its readings are intended to reflect core temperature.', 'size_i': 1360, 'name_s': 'body', 'disease_ents': [cardiac surgery], 'chemical_ents': [], 'genetic_ents': []}\n",
      "1900 paragraphs processed\n",
      "{'section_s': 'blood pressure, cholesterol, c-reactive protein, and homocysteine', 'id': '003a3ad28c0b73a0c4d8fd7eee20c955', 'article_id_s': '2281cf6e33237b219a1f33c25a7e6b12dec14619', 'text_t': 'The specific cause of most cases of hypertension is unknown. A small share of the cases is caused by kidney disease, glandular tumors, diabetes, and some drugs (including some medicines for colds and migraine), but most cases are associated with a poor lifestyle, that is, physical inactivity, unhealthy diet, stress, smoking, and ingesting too much salt and alcohol. There is a wide array of medications for treating high blood pressure but, in treating the condition, elimination of the causes and predisposing factors enumerated above is usually considered first.', 'size_i': 566, 'name_s': 'blood pressure, cholesterol, c-reactive protein, and homocysteine', 'disease_ents': [hypertension, kidney disease, glandular tumors, diabetes, migraine, high], 'chemical_ents': [alcohol], 'genetic_ents': []}\n",
      "2000 paragraphs processed\n",
      "{'section_s': 'introduction', 'id': '003d5e97e1af3975248f5df4948a8325', 'article_id_s': '741ad67bb03111d1e1c85c77b0c7f07b8137dd7a', 'text_t': 'Although ML-7 has never been used in vivo as an antiviral compound, it has been administered to rats and mice to preserve lung or intestinal epithelial barriers during stress-or lipopolysaccharideinduced injury [8] [9] [10] [11] . MLCK is a possible target for drugs intended for relieve of lung or bowel injury during sepsis because it controls paracellular permeability [12, 13] .', 'size_i': 382, 'name_s': 'introduction', 'disease_ents': [injury, lung or bowel injury, sepsis], 'chemical_ents': [ML-7], 'genetic_ents': [MLCK]}\n",
      "2100 paragraphs processed\n",
      "{'section_s': 'outbreak control', 'id': '004082ebd27973e14c302731d064606d', 'article_id_s': '1aa2cc6f509964807d510b48a63180e09ee083de', 'text_t': 'Outbreak surveillance and control should be considered a high priority for ICPs. The leading causes of LTCF outbreaks are discussed above and listed in Table 1 . When the number of cases exceeds the normal baseline, an outbreak within the facility should be considered. The ICP is advised, and required by CMS, to monitor resident and staff illnesses, since healthy personnel may acquire and transmit virulent pathogens.', 'size_i': 420, 'name_s': 'outbreak control', 'disease_ents': [LTCF], 'chemical_ents': [], 'genetic_ents': []}\n",
      "2200 paragraphs processed\n",
      "{'section_s': 'antimicrobial resistance of salmonella', 'id': '0043ecf39ef6fae03eb365faba7c856a', 'article_id_s': '6389e5fa6ae2cd574240a36bb33a9bfb92235f9f', 'text_t': 'The prevalence of STM and SCS isolates from pigs and humans that are fluoroquino lone and multidrug resistant has increased in Taiwan, and the isolates have become wide spread across the country. The SCS isolates from humans and pigs were closely related genotypically, suggesting the nationwide dis semination of the organism from pigs to humans.', 'size_i': 347, 'name_s': 'antimicrobial resistance of salmonella', 'disease_ents': [STM, SCS, SCS], 'chemical_ents': [], 'genetic_ents': []}\n",
      "2300 paragraphs processed\n",
      "{'section_s': 'introduction', 'id': '004681bdea3b0c4f5c9a771a1d42af04', 'article_id_s': 'c7a5eb7018ef7ea20d8ef8191e8c82b97471f5a9', 'text_t': 'The measurement of core temperature (T C ) is fundamental in human thermoregulation studies under resting, exercise and postexercise conditions (Cuddy et al., 2014; West et al., 2014; Xu et al., 2013) . Small T C increases result in specific physiological responses such as increased cutaneous vasodilation and sweating, which helps to dissipate excess body heat during exercise (Romanovsky, 2014; Sawka et al., 2011) . Research into obtaining a fast, valid, reliable and noninvasive measure of core temperature is of interest in many disciplinary fields. For example, during the 2009 outbreak of severe acute respiratory syndrome (SARS) generated by Influenza A (H1N1) in Southeast Asia, thermal images obtained through infrared thermography (IRT) were successfully used for screening potential infected individuals at international airports, becoming an important primary step for mass prevention of the spread of SARS (Chan et al., 2004; Lahiri et al., 2012; Ng et al., 2004; Ring and Ammer, 2012; Teunissen and Daanen, 2011) . Furthermore, occupational and sports medicine research has attempted to determine a non-invasive proxy for core temperature particularly when access to participants is limited and thermal safety is of a concern due to protective encapsulating clothing (Montain et al., 1994) , hot ambient environments (Taylor, 2006) and/or high endogenous heat production during athletic competition (Laursen et al., 2006) .', 'size_i': 1439, 'name_s': 'introduction', 'disease_ents': [severe acute respiratory syndrome, SARS, Influenza A, SARS], 'chemical_ents': [], 'genetic_ents': []}\n",
      "2400 paragraphs processed\n",
      "{'section_s': 'negev virus (50)', 'id': '004a5a6ca398d2260e4ca8530001c276', 'article_id_s': '8176a764d5a3432a6bd152a3865016d121623a3f', 'text_t': 'Forecariah virus (42) Culicoides spp.', 'size_i': 37, 'name_s': 'negev virus (50)', 'disease_ents': [], 'chemical_ents': [], 'genetic_ents': []}\n",
      "2500 paragraphs processed\n",
      "{'section_s': 'erysipelas', 'id': '004ddb1283ed7060ca826d85a6836d17', 'article_id_s': '20c6ef6f99ef5cdd70a64ad74892bbbb90a5e873', 'text_t': 'Erysipelas, caused by Erysipelothrix rhusiopathiae, which affects a variety of fishes and mammals, including man, was first recognized when Koch discovered an organism he called', 'size_i': 177, 'name_s': 'erysipelas', 'disease_ents': [Erysipelas, Erysipelothrix rhusiopathiae], 'chemical_ents': [], 'genetic_ents': []}\n",
      "2600 paragraphs processed\n",
      "{'section_s': 'parasites', 'id': '0050ac9b9baac42ba144b32f86fa43aa', 'article_id_s': '45694460eb735606f756b5850caced2cd8b313d9', 'text_t': 'Tachyzoites of T. gondii RH strain were maintained in ICR mice by intraperitoneal (i.p.) serial passage at regular 72 h intervals. The parasites were washed three times by phosphate-buffered saline (PBS) and centrifuged at 1000g for 5 min as previously described (Mack and McLeod, 1984) .', 'size_i': 288, 'name_s': 'parasites', 'disease_ents': [], 'chemical_ents': [phosphate], 'genetic_ents': []}\n",
      "2700 paragraphs processed\n",
      "{'section_s': 'western blot analysis of recombinant proteins', 'id': '0053a926677872dbb78d0d86a98a818f', 'article_id_s': '8607aac9046864becdec9a89df48524d61192e09', 'text_t': \"CEF and HeLa cells were infected with recombinant MVA-NiVsG and MVA-NiV-G viruses at a MOI of 5. Cells infected with MVA (MOI 5) and inoculation medium alone (mock) were used as controls. Cell lysates were prepared or culture supernatants were collected at various time points after infection and stored at −80 • C. Cell-associated and secreted proteins were resolved by sodium dodecyl sulfate (SDS)-polyacrylamide (10%) gel electrophoresis (SDS-PAGE), and proteins were transferred onto nitrocellulose membranes by wet electroblotting. To investigate the glycosylation pattern, proteins were pretreated using the Protein Deglycosylation Mix II kit (New England Biolabs, Ipswich, MA, USA) according to the manufacturer's instructions prior to SDS PAGE. Membranes were blocked with blocking buffer, which consisted of PBS containing 5% non-fat dried milk powder (Carl Roth, Karlsruhe, Germany) and 0.1% v/v Tween 20 (Sigma-Aldrich) for one hour at room temperature. Membranes were then probed with the primary antibody (polyclonal mouse anti-NiV-G (1:2000) or polyclonal rabbit anti-NiV-G (1:5000)) diluted in blocking buffer overnight at 4 • C. Membranes were washed three times with PBS containing 0.1% Tween 20 (PBS/T) and probed with goat anti-mouse IgG conjugated to horseradish peroxidase (HRP) (1:5000; Agilent Dako, Glostrup, Denmark) or goat anti-rabbit IgG HRP (1:5000; Cell Signaling Technology, Leiden, The Netherlands). Membranes were washed again with PBS/T and were developed using SuperSignal ® West Dura Extended Duration substrate (Thermo Fisher Scientific, Planegg, Gemany). Blots were visualized using a MicroChemi 4.2 imager (DNR Bio-Imaging Systems, Neve Yamin, Israel).\", 'size_i': 1691, 'name_s': 'western blot analysis of recombinant proteins', 'disease_ents': [], 'chemical_ents': [sodium dodecyl sulfate, Tween 20, Tween 20], 'genetic_ents': [polyclonal mouse anti-NiV-G, anti-NiV-G, goat anti-mouse IgG, horseradish peroxidase, HRP, goat anti-rabbit IgG, HRP]}\n",
      "2800 paragraphs processed\n",
      "{'section_s': 'symptome und befunde', 'id': '0056c59cafd03e04806826b3f2c7d3fc', 'article_id_s': 'ddbd4ea9d0479ae43dd09deca6b21592958c811d', 'text_t': 'Beim Immunkompetenten verläuft die Erkrankung meist asymptomatisch oder klinisch wie eine infektiöse Mononukleose oder eine Virushepatitis. Schwere Verläufe sind meist assoziiert mit einer gestörten zellulären Immunität, z. B. bei HIV-Infektion oder nach Organtransplantation. Insbesondere nach Lebertransplantationen kann es zu schweren Hepatitiden mit fulminantem Leberversagen kommen.', 'size_i': 387, 'name_s': 'symptome und befunde', 'disease_ents': [Virushepatitis], 'chemical_ents': [], 'genetic_ents': []}\n",
      "2900 paragraphs processed\n",
      "{'section_s': 'clinical features of dhf/dss and severe dengue ::: results', 'id': '0059f620048f423750dd7a653b955084', 'article_id_s': 'PMC5036306', 'text_t': 'The ROC curves (Figs. 2 and 3) illustrate the accuracy of these various models in predicting concurrent diagnosis of laboratory-confirmed DHF/DSS or SD at hospital admission, based on the total number of WS included in each model. For DHF/DSS, the highest predictive value was obtained when patients presented two or more WS upon admission (AUC = 0.78). For SD, predictive value was maximized when patients presented with at least three WS at admission (AUC = 0.81).', 'size_i': 466, 'name_s': 'clinical features of dhf/dss and severe dengue ::: results', 'disease_ents': [DHF, DSS, SD, DHF, DSS, SD], 'chemical_ents': [], 'genetic_ents': []}\n",
      "3000 paragraphs processed\n",
      "{'section_s': 'plasma lipidomic profile in cows during the transition period and in cows with fatty liver ::: food animal', 'id': '005d1fd4474162b5f39458ef865c79eb', 'article_id_s': 'PMC4895374', 'text_t': 'In a study on 12 clinical healthy cows, during their transition period, 30 masses could be analyzed by MS/MS as potential biomarkers for determination of changes within the lipidome relative to calving. The main lipid groups detected, were triacylglycerides, phosphatidylcholines, and lysophosphatidylcholines.', 'size_i': 310, 'name_s': 'plasma lipidomic profile in cows during the transition period and in cows with fatty liver ::: food animal', 'disease_ents': [], 'chemical_ents': [triacylglycerides, phosphatidylcholines, lysophosphatidylcholines], 'genetic_ents': []}\n",
      "3100 paragraphs processed\n",
      "{'section_s': 'building a predictive model to assess conservation', 'id': '0060197967278deee0088b62c766fba4', 'article_id_s': 'ccaafc13b2c1551dce74e1a7fe3e43cea5228732', 'text_t': \"A critical question regarding the use of model systems is how well the experimental model represents the target system. Calu3 cells provide a convenient in vitro model for lung epithelium, but it is unknown to what extent the pathways and mechanisms found in the in vitro model will translate to a more realistic model, such as primary cells or in vivo animal models. In order to demonstrate that our approach based on in vitro data provides predictive insights relevant to an in vivo model, we used the Inferelator software [31] to determine how well models derived from Calu3 data would apply to data from other experimental systems [18] (steps 3 and 4 in Figure 2 ). The Inferelator uses multivariate regression to select a small set of regulators that are the most likely to be influencing a regulatory target. Expression data from the Calu-3 cells were collapsed using hierarchical clustering, such that Table 1 . SARS genes whose topology is significantly altered by proteome data integration. clusters of genes with similar expression profiles were chosen as regulatory ''targets''. A list of candidate regulators was supplied to the Inferelator software as input, which was compiled from known transcription factors and the top candidate regulators from our ranking strategy [32] . A prediction of the critical regulators of a gene cluster can be used to predict the behavior of that gene cluster based on the behavior of its predicted regulators. Behavioral conservation can therefore be evaluated by comparing predicted cluster expression profiles with expression data in a distinct, but comparable biological system. We applied the SARS-CoV model to a dataset of SARS-CoV infection of primary human airway epithelial (HAE) cells. These cultures are an in vitro model of the human lung that morphologically and physiologically recapitulates the epithelium of the conducting airway and have been demonstrated to be permissive for SARS-CoV infection [33] . The cultures differentiate from primary cells into ciliated, goblet (mucin-producing), and basement membrane cells in a stratified epithelium. For influenza data, the Calu3 model was fitted to a set of mouse influenza infection experiments using VN1203 to infect mice with 10 3 , 10 4 , or 10 5 PFU at 1 and 2 days post-infection [18] . Calu3 models were applied to new data (HAE for SARS-CoV, mouse for influenza virus) by using the expression levels of the input regulators (measured in the new system) to calculate the expression level of each gene cluster in the new system, using the model learned from the Calu3 experimental system. Figures 7A  and B show the correlation of Calu3 models' predicted output of each gene cluster with the observed output of the actual in vivo or ex vivo data. Since the SARS-CoV Calu3/HAE comparison is within the human system and within a cell culture context, while the influenza Calu3/mouse comparison spans distinct species and model system types, these comparisons from the two virus types yielded somewhat different results. Unsurprisingly, overall, the Calu3-based model of SARS-CoV infection was more compatible with HAE data than was the Calu3 model of influenza infection with the mouse data. However, assessing agreement for individual clusters showed that 6 out of 28 clusters yielded correlation above 0.8 in SARS-CoV, while 9 out of 30 clusters had correlation above this level in influenza virus. As an illustration, the expression profile of a single high performing cluster prediction from each virus is included for comparison with the observed expression of the cluster (Figure 7b ). The high performing SARS-CoV clusters were enriched for genes involved in the immune response. The two highest performing influenza clusters were also enriched for immune response genes, while another showed enrichment for transcription regulation. For each virus type, we also generated 100 models from datasets with scrambled genes but with the cluster structure intact. In this way, we compared the observed level of cross-model fitting with the level of model fitting expected by random chance. Figure 7c shows the average Pearson's correlation of the clusters' predicted profiles and the observed profiles from the new data. The average correlation of models derived from data with scrambled rows is shown for comparison. While these results demonstrate the high degree of regulatory conservation between the different models, we decided to identify the individual genes that displayed the most conserved response to infection across model systems. To this end, we used the crossprediction metric (Xpred) described in [18] , which is a measure of how well the behavior of a gene is predicted across model systems. Xpred rankings are shown in Table S3 (SARS-CoV) and Table S4 (influenza virus). To determine if these genes showed enrichment in functional categories, we used GO term enrichment analysis on genes with the highest Xpred scores in each of the two virus datasets. For both viruses, genes with a conserved response showed functional enrichment in innate immune response signaling (not shown). Thus while not all regulatory mechanisms are conserved between model systems, a component of the innate immune response is well-preserved.\", 'size_i': 5252, 'name_s': 'building a predictive model to assess conservation', 'disease_ents': [SARS-CoV, SARS-CoV infection, SARS-CoV infection, SARS-CoV, SARS-CoV, SARS-CoV infection, SARS-CoV, SARS-CoV, SARS-CoV], 'chemical_ents': [], 'genetic_ents': [SARS genes, mucin]}\n",
      "3200 paragraphs processed\n",
      "{'section_s': 'complications of percutaneous endoscopic gastrostomy in dogs and cats receiving corticosteroid therapy: a retrospective case‐control study ::: esvcn‐o‐1', 'id': '006308ac963e6877910f355392854bda', 'article_id_s': 'PMC4913621', 'text_t': '\\nJ. Aguiar, Y.M. Chang, O. Garden.\\n', 'size_i': 35, 'name_s': 'complications of percutaneous endoscopic gastrostomy in dogs and cats receiving corticosteroid therapy: a retrospective case‐control study ::: esvcn‐o‐1', 'disease_ents': [], 'chemical_ents': [], 'genetic_ents': []}\n",
      "3300 paragraphs processed\n",
      "{'section_s': 'total co 2', 'id': '0065da51e77a4f1941772953c822e268', 'article_id_s': '36f0ce11b61edc9a0eb039d08bc0e0d86b5f575b', 'text_t': '] is calculated using the metabolic acidosis. Obstruction of the upper intestinal tract, or abomasal stasis, is characterized by varying degrees of dehydration and metabolic alkalosis with hypochloremia and hypokalemia. Chronic renal disease is characterized by hyponatremia and hypochloremia. Chronic inappetence in herbivores is characterized by hypokalemia, particularly in lactating ruminants. A combination of the clinical assessment and the available laboratory evaluation will allow the clinician to make the most rational approach to treatment.', 'size_i': 552, 'name_s': 'total co 2', 'disease_ents': [metabolic acidosis, Obstruction of the upper intestinal tract, abomasal stasis, dehydration, metabolic alkalosis, hypochloremia, hypokalemia, Chronic renal disease, hyponatremia, hypochloremia, Chronic inappetence, hypokalemia], 'chemical_ents': [], 'genetic_ents': []}\n",
      "3400 paragraphs processed\n",
      "{'section_s': 'single molecule real-time sequencing', 'id': '006951c8bdba55fd91fe357680ecfd7d', 'article_id_s': '6bed996a1f77c771e5c32d168eeef1f88eb8553a', 'text_t': 'Currently Pacific Biosciences offers two devices, the Sequel System and the PacBio RS II [Ari and Arikan, 2016; Heather and Chain, 2016; Kulski, 2016; Liu et al., 2012; Metzker, 2010 ; ,http://allseq.com/knowledge-bank/sequencing-platforms/pacific-biosciences/. (accessed: 2018.09.22)].', 'size_i': 286, 'name_s': 'single molecule real-time sequencing', 'disease_ents': [], 'chemical_ents': [], 'genetic_ents': []}\n",
      "3500 paragraphs processed\n",
      "{'section_s': 'discussion', 'id': '006c3ba7dcdd77fe502c5385dc8638f5', 'article_id_s': 'b2b148720d1cf79fbeac9a409962bf9e2e190c4a', 'text_t': 'Human T-lymphotropic virus type-1 infection causes several organ-specific inflammatory diseases including HAM/TSP (2, 3) . Previous reports demonstrating that HTLV-1 proviral loads are high in affected organs such as the muscles, lungs, and CNS suggest that HTLV-1Yinfected cells accumulate in the organs (13, 35) . The pathogenesis model in which both HTLV-1Yinfected CD4-positive T cells and the virus-specific CD8-positive CTLs infiltrate the organs from the peripheral blood followed by bystander tissue damage may explain why HTLV-1 infection can cause several chronic inflammatory diseases in various organs. Further studies are needed to determine whether the similar immunopathologic model can be applied to HTLV-1Yassociated inflammatory diseases in other organs. ', 'size_i': 773, 'name_s': 'discussion', 'disease_ents': [T-lymphotropic virus, organ-specific, inflammatory diseases, HAM, TSP, HTLV-1 infection, chronic inflammatory diseases, HTLV-1Yassociated inflammatory diseases], 'chemical_ents': [], 'genetic_ents': []}\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-45-713aa61dc04c>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mCORD19Processor\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mwindow_size\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m200\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m<ipython-input-44-a8ea34b24b94>\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, window_size)\u001b[0m\n\u001b[1;32m      4\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcovid_paragraphs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpysolr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSolr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'http://librairy.linkeddata.es/data/covid-paragraphs'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m10\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwindow_size\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mwindow_size\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 6\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mannotate_paragraphs\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      7\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      8\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-44-a8ea34b24b94>\u001b[0m in \u001b[0;36mannotate_paragraphs\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m     46\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     47\u001b[0m                 \u001b[0;32mfor\u001b[0m \u001b[0mparagraph\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mparagraphs\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 48\u001b[0;31m                     \u001b[0mdoc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnlp\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mparagraph\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'text_t'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     49\u001b[0m                     \u001b[0mparagraph\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'disease_ents'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mf\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mf\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mdoc\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0ments\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlabel_\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'DISEASE'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     50\u001b[0m                     \u001b[0mparagraph\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'chemical_ents'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mf\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mf\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mdoc\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0ments\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlabel_\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'CHEMICAL'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/spacy/language.py\u001b[0m in \u001b[0;36m__call__\u001b[0;34m(self, text, disable, component_cfg)\u001b[0m\n\u001b[1;32m    988\u001b[0m                 \u001b[0merror_handler\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mproc\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_error_handler\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    989\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 990\u001b[0;31m                 \u001b[0mdoc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mproc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdoc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mcomponent_cfg\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    991\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    992\u001b[0m                 \u001b[0;31m# This typically happens if a component is not initialized\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Master/TFM/bio-nlp/NER_processing.py\u001b[0m in \u001b[0;36m__call__\u001b[0;34m(self, doc)\u001b[0m\n\u001b[1;32m     61\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__call__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdoc\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mDoc\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m->\u001b[0m \u001b[0mDoc\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     62\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mentities\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mclass_entities\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mEntities\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdoc\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 63\u001b[0;31m         \u001b[0mprocess_by_paragraph\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdoc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mentities\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     64\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mentities\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpostprocessing\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     65\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mdoc\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Master/TFM/bio-nlp/NER_processing.py\u001b[0m in \u001b[0;36mprocess_by_paragraph\u001b[0;34m(doc, entities)\u001b[0m\n\u001b[1;32m     29\u001b[0m         \u001b[0;31m# print(len(str(paragraph)))\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     30\u001b[0m         \u001b[0mdisease_service\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msentence_to_process\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mparagraph\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 31\u001b[0;31m         \u001b[0mdisease_results\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdisease_service\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpredict\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     32\u001b[0m         \u001b[0mentities\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend_new_entities\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdisease_results\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     33\u001b[0m         \u001b[0mchemical_service\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msentence_to_process\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mparagraph\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Master/TFM/bio-nlp/bioprocessor.py\u001b[0m in \u001b[0;36mpredict\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m     36\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     37\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mpredict\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 38\u001b[0;31m         \u001b[0mresults\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpipeline\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msequence\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     39\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moffset\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     40\u001b[0m             \u001b[0;32mfor\u001b[0m \u001b[0mresult\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mresults\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/transformers/pipelines/token_classification.py\u001b[0m in \u001b[0;36m__call__\u001b[0;34m(self, inputs, **kwargs)\u001b[0m\n\u001b[1;32m    174\u001b[0m                     \u001b[0;32mwith\u001b[0m \u001b[0mtorch\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mno_grad\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    175\u001b[0m                         \u001b[0mtokens\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mensure_tensor_on_device\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m**\u001b[0m\u001b[0mtokens\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 176\u001b[0;31m                         \u001b[0mentities\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmodel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m**\u001b[0m\u001b[0mtokens\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcpu\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnumpy\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    177\u001b[0m                         \u001b[0minput_ids\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtokens\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"input_ids\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcpu\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnumpy\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    178\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_call_impl\u001b[0;34m(self, *input, **kwargs)\u001b[0m\n\u001b[1;32m    887\u001b[0m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_slow_forward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    888\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 889\u001b[0;31m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mforward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    890\u001b[0m         for hook in itertools.chain(\n\u001b[1;32m    891\u001b[0m                 \u001b[0m_global_forward_hooks\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/transformers/models/bert/modeling_bert.py\u001b[0m in \u001b[0;36mforward\u001b[0;34m(self, input_ids, attention_mask, token_type_ids, position_ids, head_mask, inputs_embeds, labels, output_attentions, output_hidden_states, return_dict)\u001b[0m\n\u001b[1;32m   1677\u001b[0m         \u001b[0mreturn_dict\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mreturn_dict\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0mreturn_dict\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m \u001b[0;32melse\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mconfig\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0muse_return_dict\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1678\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1679\u001b[0;31m         outputs = self.bert(\n\u001b[0m\u001b[1;32m   1680\u001b[0m             \u001b[0minput_ids\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1681\u001b[0m             \u001b[0mattention_mask\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mattention_mask\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_call_impl\u001b[0;34m(self, *input, **kwargs)\u001b[0m\n\u001b[1;32m    887\u001b[0m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_slow_forward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    888\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 889\u001b[0;31m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mforward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    890\u001b[0m         for hook in itertools.chain(\n\u001b[1;32m    891\u001b[0m                 \u001b[0m_global_forward_hooks\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/transformers/models/bert/modeling_bert.py\u001b[0m in \u001b[0;36mforward\u001b[0;34m(self, input_ids, attention_mask, token_type_ids, position_ids, head_mask, inputs_embeds, encoder_hidden_states, encoder_attention_mask, past_key_values, use_cache, output_attentions, output_hidden_states, return_dict)\u001b[0m\n\u001b[1;32m    969\u001b[0m             \u001b[0mpast_key_values_length\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mpast_key_values_length\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    970\u001b[0m         )\n\u001b[0;32m--> 971\u001b[0;31m         encoder_outputs = self.encoder(\n\u001b[0m\u001b[1;32m    972\u001b[0m             \u001b[0membedding_output\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    973\u001b[0m             \u001b[0mattention_mask\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mextended_attention_mask\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_call_impl\u001b[0;34m(self, *input, **kwargs)\u001b[0m\n\u001b[1;32m    887\u001b[0m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_slow_forward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    888\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 889\u001b[0;31m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mforward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    890\u001b[0m         for hook in itertools.chain(\n\u001b[1;32m    891\u001b[0m                 \u001b[0m_global_forward_hooks\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/transformers/models/bert/modeling_bert.py\u001b[0m in \u001b[0;36mforward\u001b[0;34m(self, hidden_states, attention_mask, head_mask, encoder_hidden_states, encoder_attention_mask, past_key_values, use_cache, output_attentions, output_hidden_states, return_dict)\u001b[0m\n\u001b[1;32m    566\u001b[0m                 )\n\u001b[1;32m    567\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 568\u001b[0;31m                 layer_outputs = layer_module(\n\u001b[0m\u001b[1;32m    569\u001b[0m                     \u001b[0mhidden_states\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    570\u001b[0m                     \u001b[0mattention_mask\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_call_impl\u001b[0;34m(self, *input, **kwargs)\u001b[0m\n\u001b[1;32m    887\u001b[0m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_slow_forward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    888\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 889\u001b[0;31m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mforward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    890\u001b[0m         for hook in itertools.chain(\n\u001b[1;32m    891\u001b[0m                 \u001b[0m_global_forward_hooks\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/transformers/models/bert/modeling_bert.py\u001b[0m in \u001b[0;36mforward\u001b[0;34m(self, hidden_states, attention_mask, head_mask, encoder_hidden_states, encoder_attention_mask, past_key_value, output_attentions)\u001b[0m\n\u001b[1;32m    494\u001b[0m             \u001b[0mpresent_key_value\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpresent_key_value\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mcross_attn_present_key_value\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    495\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 496\u001b[0;31m         layer_output = apply_chunking_to_forward(\n\u001b[0m\u001b[1;32m    497\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfeed_forward_chunk\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mchunk_size_feed_forward\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mseq_len_dim\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mattention_output\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    498\u001b[0m         )\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/transformers/modeling_utils.py\u001b[0m in \u001b[0;36mapply_chunking_to_forward\u001b[0;34m(forward_fn, chunk_size, chunk_dim, *input_tensors)\u001b[0m\n\u001b[1;32m   1815\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mtorch\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0moutput_chunks\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdim\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mchunk_dim\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1816\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1817\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mforward_fn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0minput_tensors\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/transformers/models/bert/modeling_bert.py\u001b[0m in \u001b[0;36mfeed_forward_chunk\u001b[0;34m(self, attention_output)\u001b[0m\n\u001b[1;32m    506\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    507\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mfeed_forward_chunk\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mattention_output\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 508\u001b[0;31m         \u001b[0mintermediate_output\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mintermediate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mattention_output\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    509\u001b[0m         \u001b[0mlayer_output\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moutput\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mintermediate_output\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mattention_output\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    510\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mlayer_output\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/torch/nn/modules/module.py\u001b[0m in \u001b[0;36m_call_impl\u001b[0;34m(self, *input, **kwargs)\u001b[0m\n\u001b[1;32m    887\u001b[0m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_slow_forward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    888\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 889\u001b[0;31m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mforward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    890\u001b[0m         for hook in itertools.chain(\n\u001b[1;32m    891\u001b[0m                 \u001b[0m_global_forward_hooks\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/transformers/models/bert/modeling_bert.py\u001b[0m in \u001b[0;36mforward\u001b[0;34m(self, hidden_states)\u001b[0m\n\u001b[1;32m    410\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mforward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhidden_states\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    411\u001b[0m         \u001b[0mhidden_states\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdense\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhidden_states\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 412\u001b[0;31m         \u001b[0mhidden_states\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mintermediate_act_fn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhidden_states\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    413\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mhidden_states\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    414\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/biobert/lib/python3.8/site-packages/torch/nn/functional.py\u001b[0m in \u001b[0;36mgelu\u001b[0;34m(input)\u001b[0m\n\u001b[1;32m   1457\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mhas_torch_function_unary\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1458\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mhandle_torch_function\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgelu\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0minput\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1459\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mtorch\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_C\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_nn\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgelu\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1460\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1461\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "CORD19Processor(window_size=200)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "502792c7-f153-4289-b904-b3cce07fea6f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "19a3b91f-52eb-4c67-9649-339a854d93c8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1000 paragraphs retrieved\n",
      "100 paragraphs processed\n",
      "{'section_s': 'subunit vaccines', 'id': '43c7274ceb6399f07a7e0a52bbcd30ba', 'article_id_s': 'a9754e66d1d118157221031daab60604bab2ad14', 'text_t': 'Both MERS-CoV and SARS-CoV RBDs consist of a core and a receptor-binding subdomain. They share a high degree of structural similarity in the core subdomains, but the receptor-binding subdomains are notably divergent [32] . Because of variation of the receptor-binding subdomain region within subgroups of the same CoV or across different CoV groups, it is advisable to rationally design MERS subunit vaccines using chimeric S protein containing several neutralizing epitopes from divergent subgroups. This approach will provide a strategic platform for the rational design of subunit vaccines against future emerging CoVs by focusing on the chimeric S protein containing neutralizing epitopes from multiple virus strains across different subgroups [94] .', 'size_i': 754, 'disease_ents': [MERS-CoV, SARS-CoV, MERS], 'chemical_ents': [], 'genetic_ents': [S protein, S protein]}\n",
      "200 paragraphs processed\n",
      "{'section_s': 'collaborative accountability in ict-based public health communication', 'id': '9f1859a6a305d9b1a0a0853ccfde412c', 'article_id_s': '340dfd53fcc624ce7c200694b406cd40b92761ba', 'text_t': 'In the context of health, how are risks assessed and addressed? Generally speaking, risks are managed by following a prescribed process (Crouhy, Galai, & Mark, 2006 ) that includes the following steps:', 'size_i': 201, 'disease_ents': [], 'chemical_ents': [], 'genetic_ents': []}\n",
      "300 paragraphs processed\n",
      "{'section_s': 'virus structure and replication', 'id': '589bb570f19cb0b43d0df85f26eaf4ec', 'article_id_s': '44232419011da5172b4e950153f8fe563563e7e2', 'text_t': 'The virus assembly process is not understood completely, but available data indicate that viral nucleocapsid associates with regions of the plasma membrane enriched with M and G proteins. The nucleocapsid then condenses into a form that is transcriptionally inactive, and in a process that appears to be primarily driven by the M protein [36, 48] , the plasma membrane containing G protein envelops the underlying viral nucleocapsid and mature virus particles bud from the cell surface. The G protein cytoplasmic tail (CT) does not appear to have an essential role in virus budding [85] , but does enhance efficiency of the process [97, 98] . The mature virus particles contain the viral RNA polymerase necessary for further propagation in susceptible cells.', 'size_i': 758, 'disease_ents': [], 'chemical_ents': [], 'genetic_ents': [M, G proteins, M protein, G protein, G protein cytoplasmic tail, CT, viral RNA polymerase]}\n",
      "400 paragraphs processed\n",
      "{'section_s': 'cell, virus and animal', 'id': 'f7075e4e666a31710f8df05091f5e456', 'article_id_s': '23bfd5b47a8f2b51410c7c9ccc33981b8d200657', 'text_t': 'De novo synthesis, construction, expression and purification of recombinant GRFT Five pairs of primers were designed (Table 1 ) based on the GRFT sequence (GenBank ID: FJ594069.1) to synthesize the GRFT gene de novo by the splicing by overlap extension PCR (SOE-PCR) method [7] . The full-length GRFT PCR product (375 bp) was digested with NdeI and XhoI restriction enzymes and ligated into pCold-I vector (Takara Bio Inc.) that had been digested with the same enzymes. The construct was confirmed by restriction enzyme digestion and DNA sequencing analysis.', 'size_i': 558, 'disease_ents': [primers], 'chemical_ents': [], 'genetic_ents': [recombinant GRFT, GRFT sequence, GRFT gene, NdeI, XhoI restriction enzymes]}\n",
      "500 paragraphs processed\n",
      "{'section_s': 'body', 'id': '7b6a3c05c8ec4544c28816e7341a0a9c', 'article_id_s': '4d3a4af185bf33f6542c99ed52bb670a72cb1d80', 'text_t': 'During a 3-year period (2003) (2004) (2005) , high-resolution CT scans of the lungs of 6 HSCT recipients who developed PIV3 pneumonia at Westmead Hospital, a quaternary referral center for HSCT, were made available. On the basis of these highresolution CT scans, we now report that multiple small nodules are common in the lungs of HSCT recipients with PIV3 pneumonia.', 'size_i': 368, 'disease_ents': [PIV3 pneumonia, PIV3 pneumonia], 'chemical_ents': [], 'genetic_ents': []}\n",
      "600 paragraphs processed\n",
      "{'section_s': 'pegylated ifn-α', 'id': '398d1d3246647884935158912b80543e', 'article_id_s': '0b99f04241566a49e266f6b6848f7714b7aaad05', 'text_t': 'Two forms of pegylated IFN-α have been approved by the Food and Drug Administration (FDA) and by the European Medicines Agency (EMEA) for the ', 'size_i': 142, 'disease_ents': [], 'chemical_ents': [pegylated IFN-α], 'genetic_ents': []}\n",
      "700 paragraphs processed\n",
      "{'section_s': 'sa-mcf occurs where bovids and deer come in contact with sheep.', 'id': '3cd6af64340579cea27d7b5dabfe5f48', 'article_id_s': '7aa1630daebbd7ec1845b4dbf03978b53a3903fa', 'text_t': 'Gross morbid changes may not be present in occasional animals that die of peracute MCF, and in these the diagnosis must rest on the detection of the characteristic histologic changes, and demonstration of the genome of an implicated rhadinovirus in tissue.', 'size_i': 256, 'disease_ents': [peracute, MCF], 'chemical_ents': [], 'genetic_ents': []}\n",
      "800 paragraphs processed\n",
      "{'section_s': 'example 5.4.', 'id': '2158228b5f143a04c43ade97f6b2f5be', 'article_id_s': '0b70c1fd82bd1962ab7eec83bdc61fcdb9a2db56', 'text_t': 'Consider the system, 5.6. Example 5.6.', 'size_i': 38, 'disease_ents': [], 'chemical_ents': [], 'genetic_ents': []}\n",
      "900 paragraphs processed\n",
      "{'section_s': 'differential diagnosis', 'id': '81ef7b9bc96e084aa6b1e683fa7dcb0a', 'article_id_s': 'c6c3363dde567367ebc8c399d7dc87303ef15aa7', 'text_t': 'Hemoglobinuria differentiates neonatal isoerythrolysis from other causes of early fading kitten syndrome.', 'size_i': 105, 'disease_ents': [Hemoglobinuria, neonatal isoerythrolysis, kitten syndrome], 'chemical_ents': [], 'genetic_ents': []}\n",
      "1000 paragraphs processed\n",
      "{'section_s': 'discussion', 'id': '88aefa8970bb9336ab93007a7f136986', 'article_id_s': '268a89879b470523856211ef78dd1e375f7a48e6', 'text_t': 'In the present study we demonstrated that rhinitis reduced the mean peak plasma nicotine concentration and significantly prolonged the time to peak concentration of intranasally administered nicotine. Pretreatment with xylometazoline further prolonged the absorption time in comparison with untreated rhinitis without affecting the peak plasma nicotine level. As the extent of absorption was largely unaffected, the data indicate slowing of the absorption process. The reduced rate of nicotine absorption during the common cold was probably the result of slower access of nicotine to the subepithelial capillaries in the well vascularised nasal mucosa [12] , and the further reduction found after pretreatment with the vasoconstrictor was the result of a decrease in the nasal mucosal blood flow [13] .', 'size_i': 802, 'disease_ents': [rhinitis, rhinitis], 'chemical_ents': [nicotine, nicotine, xylometazoline, nicotine, nicotine, nicotine], 'genetic_ents': []}\n"
     ]
    }
   ],
   "source": [
    "process_cord19()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "aaecfbeb-3d61-43b3-8e4c-177867f1a98d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "3212e806-1ef2-4861-9a3e-a9eae3e03190",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "0\n"
     ]
    }
   ],
   "source": [
    "doc = nlp('pneumonia')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b0f36aae-44bc-4b7a-8112-2bce861e7367",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python (biobert)",
   "language": "python",
   "name": "biobert"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
